<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 3 Figure 3 | CancerMHBs</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content=" 3 Figure 3 | CancerMHBs" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 3 Figure 3 | CancerMHBs" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Zhiqiang Zhang,Yuyang Hong, Hai Fang, Jiantao Shi" />


<meta name="date" content="2024-06-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-2.html"/>
<link rel="next" href="figure-4.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://jiantaoshi.github.io/cancermhaps/index.html">CancerMHBs</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Figure 1</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cancer-mhb-calling"><i class="fa fa-check"></i><b>1.1</b> Cancer MHB Calling</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#c-e.-mhb-annotation"><i class="fa fa-check"></i><b>1.2</b> (C-E). MHB annotation</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#f.-mhb-map"><i class="fa fa-check"></i><b>1.3</b> (F). MHB Map</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#g.-cancer-mhbs-vs-normal-mhbs"><i class="fa fa-check"></i><b>1.4</b> (G). Cancer MHBs vs Normal MHBs</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#h.-correlation-of-methylation-on-the-same-mhb"><i class="fa fa-check"></i><b>1.5</b> (H). Correlation of methylation on the same MHB</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#i.-enrichment-of-mhbs-in-genomic-features."><i class="fa fa-check"></i><b>1.6</b> (I). Enrichment of MHBs in genomic features.</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>2</b> Figure 2</a>
<ul>
<li class="chapter" data-level="2.1" data-path="figure-2.html"><a href="figure-2.html#a.-cancer-mhbs-and-open-chromatin"><i class="fa fa-check"></i><b>2.1</b> (A). Cancer MHBs and open chromatin</a></li>
<li class="chapter" data-level="2.2" data-path="figure-2.html"><a href="figure-2.html#b.-cancer-mhbs-and-dnam-features"><i class="fa fa-check"></i><b>2.2</b> (B). Cancer MHBs and DNAm features</a></li>
<li class="chapter" data-level="2.3" data-path="figure-2.html"><a href="figure-2.html#c-d.-enrichment-of-cancer-mhbs-in-dnam-feature-lola"><i class="fa fa-check"></i><b>2.3</b> (C-D). Enrichment of Cancer MHBs in DNAm feature (LOLA)</a></li>
<li class="chapter" data-level="2.4" data-path="figure-2.html"><a href="figure-2.html#e.-genomic-features-with-mhbs-in-open-chromatin"><i class="fa fa-check"></i><b>2.4</b> (E). Genomic features with MHBs in open chromatin</a></li>
<li class="chapter" data-level="2.5" data-path="figure-2.html"><a href="figure-2.html#f.-enrichment-of-mhbs-in-chia-pet-in-a-disease-status-specific-manner."><i class="fa fa-check"></i><b>2.5</b> (F). Enrichment of MHBs in ChIA-PET in a disease status-specific manner.</a></li>
<li class="chapter" data-level="2.6" data-path="figure-2.html"><a href="figure-2.html#g.-enrichment-of-chia-pet-tags-in-mhbs."><i class="fa fa-check"></i><b>2.6</b> (G). Enrichment of ChIA-PET tags in MHBs.</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>3</b> Figure 3</a>
<ul>
<li class="chapter" data-level="3.1" data-path="figure-3.html"><a href="figure-3.html#a.-mhb-vs-dmr"><i class="fa fa-check"></i><b>3.1</b> (A). MHB vs DMR</a></li>
<li class="chapter" data-level="3.2" data-path="figure-3.html"><a href="figure-3.html#b-c.-enrichment-of-cancer-mhbs-in-dmr-lola"><i class="fa fa-check"></i><b>3.2</b> (B-C). Enrichment of Cancer MHBs in DMR (LOLA)</a></li>
<li class="chapter" data-level="3.3" data-path="figure-3.html"><a href="figure-3.html#d-e.-escc-dnam-and-rna-profiles-gse149608"><i class="fa fa-check"></i><b>3.3</b> (D-E). ESCC DNAm and RNA profiles (GSE149608)</a></li>
<li class="chapter" data-level="3.4" data-path="figure-3.html"><a href="figure-3.html#f-g.-tf-activities"><i class="fa fa-check"></i><b>3.4</b> (F-G). TF activities</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>4</b> Figure 4</a>
<ul>
<li class="chapter" data-level="4.1" data-path="figure-4.html"><a href="figure-4.html#b.-crc-single-cell-mhap"><i class="fa fa-check"></i><b>4.1</b> (B). CRC single-cell mHap</a></li>
<li class="chapter" data-level="4.2" data-path="figure-4.html"><a href="figure-4.html#c.-crc-single-cell-mhbs"><i class="fa fa-check"></i><b>4.2</b> (C). CRC single-cell MHBs</a></li>
<li class="chapter" data-level="4.3" data-path="figure-4.html"><a href="figure-4.html#d-single-cell-mhb-vs-bulk-mhb"><i class="fa fa-check"></i><b>4.3</b> (D) Single cell MHB vs bulk MHB</a></li>
<li class="chapter" data-level="4.4" data-path="figure-4.html"><a href="figure-4.html#e.-association-between-gene-expression-and-the-present-of-mhbs-in-promoter-regions."><i class="fa fa-check"></i><b>4.4</b> (E). Association between gene expression and the present of MHBs in promoter regions.</a></li>
<li class="chapter" data-level="4.5" data-path="figure-4.html"><a href="figure-4.html#f.-association-of-gene-expression-and-methylation-levels-of-mhb-regions"><i class="fa fa-check"></i><b>4.5</b> (F). Association of gene expression and methylation levels of MHB regions</a></li>
<li class="chapter" data-level="4.6" data-path="figure-4.html"><a href="figure-4.html#i-j.-mhbs-and-dysregulation-of-gene-expression-in-crc."><i class="fa fa-check"></i><b>4.6</b> (I-J). MHBs and dysregulation of gene expression in CRC.</a></li>
<li class="chapter" data-level="4.7" data-path="figure-4.html"><a href="figure-4.html#k.-tcga-coad-dataset-validation"><i class="fa fa-check"></i><b>4.7</b> (K). TCGA-COAD dataset Validation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>5</b> Figure 5</a>
<ul>
<li class="chapter" data-level="5.1" data-path="figure-5.html"><a href="figure-5.html#a-b.-enrichment-of-dmr-or-mhb-associated-genes-in-degs"><i class="fa fa-check"></i><b>5.1</b> (A-B). Enrichment of DMR or MHB-associated genes in DEGs</a></li>
<li class="chapter" data-level="5.2" data-path="figure-5.html"><a href="figure-5.html#c.-pathway-enrichment-of-mhb-associated-genes-in-pan-cancer"><i class="fa fa-check"></i><b>5.2</b> (C). Pathway enrichment of MHB-associated genes in pan-cancer</a></li>
<li class="chapter" data-level="5.3" data-path="figure-5.html"><a href="figure-5.html#e.-kegg-pathway-crosstalk-analysis-for-pan-cancer"><i class="fa fa-check"></i><b>5.3</b> (E). KEGG Pathway crosstalk analysis for pan-cancer</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>6</b> Figure 6</a>
<ul>
<li class="chapter" data-level="6.1" data-path="figure-6.html"><a href="figure-6.html#a-b.-aging-associate-with-the-mhb"><i class="fa fa-check"></i><b>6.1</b> (A-B). Aging associate with the MHB</a></li>
<li class="chapter" data-level="6.2" data-path="figure-6.html"><a href="figure-6.html#c.-aging-associated-mhb-in-tcga-dataset"><i class="fa fa-check"></i><b>6.2</b> (C). Aging associated mhb in TCGA dataset</a></li>
<li class="chapter" data-level="6.3" data-path="figure-6.html"><a href="figure-6.html#d-m-aging-associated-mhb-functions-in-early-cancer-detection-hm450k"><i class="fa fa-check"></i><b>6.3</b> (D-M) Aging associated MHB functions in Early Cancer Detection (HM450K)</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://jiantaoshi.github.io/cancermhaps/index.html" target="blank"> The mHap map of cancers</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CancerMHBs</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-3" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number"> 3</span> Figure 3<a href="figure-3.html#figure-3" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="Data/Figures/Figure3.png" width="80%" style="display: block; margin: auto;" />
<strong>Figure 3. MHBs are associated with dysregulation of gene expression in cancers.</strong> (A) Annotation of MHBs to DMRs. Hyper, hyper-DMR. Hypo, hypo-DMR. NC, no significant change. ND, not determined. (B-C) The enrichment of MHBs in hyper- and hypo-DMRs, respectively. Enrichment test was performed by R package LOLA, using the union of MHBs as the background. The resulting p-values for significant enrichment were ranked across all cancer types (FDR &lt; 0.01). The black rectangle highlights the top one. Nonsignificant (FDR &gt; 0.01) results are shown in grey. (D) Association of MHBs and dysregulation of gene expression in ESCC. For a dataset of esophageal squamous cell carcinoma (ESCC, GSE149608) with 10 tumor and paired 10 normal WGBS and RNA-seq samples, DMRs were identified using Metilene (FDR &lt; 0.05, Δbeta &gt; 0.1). ESCC cancer MHBs (n = 27497) were identified using mHapSuite. The identification of potentially genes regulated by MHB and DMR using rGREAT with default parameters. Meanwhile, the DEGs (n = 2769) between ESCC and adjacent normal tissue performed by wilcoxon rank sum test (Log2FC &gt;= 1, FDR &lt; 0.05). Genes without DMRs were used to construct a 2x2 contingency table that separates each gene into one of four categories based on two factors, i.e., status of MHB and differential expression. Statistical significance was evaluated by Fisher’s exact test. (E) A heatmap shows the mean methylation and expression of upregulated genes that also contains MHBs. (F) A heatmap shows the transcription factors (TF) activities between ESCC and normal tissues. The estimation of TF activities with RNA-seq by using decoupleR. The top 50 variable regulators activity difference between ESCC and normal samples were displayed. The genes involved in G2M checkpoint, Wnt signaling and P53 pathway were colored in red, green and blue, respectively. (G). TF ChIP-seq profiles from the ESCC cell line TE5 and TE10 to validate the activity of KLF5 and SOX2. MHB regions were classified as ESCC tumor-specific (n = 20778), normal-specific (n = 10501), or shared (n = 7368). Random genomic background (n = 27497) was generated by ‘bedtools shuffle’ function.</p>
<div id="a.-mhb-vs-dmr" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> (A). MHB vs DMR<a href="figure-3.html#a.-mhb-vs-dmr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb17-1"><a href="figure-3.html#cb17-1" tabindex="-1"></a><span class="co"># =============== shell command ====================</span></span>
<span id="cb17-2"><a href="figure-3.html#cb17-2" tabindex="-1"></a><span class="co"># 1. DMR calling (Metilene)</span></span>
<span id="cb17-3"><a href="figure-3.html#cb17-3" tabindex="-1"></a><span class="co"># prepare the metilene input files</span></span>
<span id="cb17-4"><a href="figure-3.html#cb17-4" tabindex="-1"></a><span class="va">tag</span><span class="op">=</span><span class="st">&quot;breast&quot;</span></span>
<span id="cb17-5"><a href="figure-3.html#cb17-5" tabindex="-1"></a><span class="ex">metilene_input.pl</span> <span class="at">--in1</span>  merged_CpG/<span class="va">${tag}</span>_T_Merged_CpG.bedGraph <span class="dt">\</span></span>
<span id="cb17-6"><a href="figure-3.html#cb17-6" tabindex="-1"></a>          <span class="at">--in2</span>  merged_CpG/<span class="va">${tag}</span>_N_Merged_CpG.bedGraph <span class="dt">\</span></span>
<span id="cb17-7"><a href="figure-3.html#cb17-7" tabindex="-1"></a>          <span class="at">--out</span>  metilene_input/<span class="va">${tag}</span>_metilene_input.bedGraph <span class="dt">\</span></span>
<span id="cb17-8"><a href="figure-3.html#cb17-8" tabindex="-1"></a>          <span class="at">--h1</span> g1_<span class="va">${tag}</span>_T <span class="dt">\</span></span>
<span id="cb17-9"><a href="figure-3.html#cb17-9" tabindex="-1"></a>          <span class="at">--h2</span> g2_<span class="va">${tag}</span>_N</span>
<span id="cb17-10"><a href="figure-3.html#cb17-10" tabindex="-1"></a><span class="co"># call DMR</span></span>
<span id="cb17-11"><a href="figure-3.html#cb17-11" tabindex="-1"></a><span class="ex">metilene_linux64</span> <span class="at">-t</span> 10 <span class="at">-c</span> 2 <span class="at">-m</span> 5 <span class="at">-a</span> g1 <span class="at">-b</span> g2 metilene_input/<span class="va">${tag}</span>_metilene_input.bedGraph <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb17-12"><a href="figure-3.html#cb17-12" tabindex="-1"></a>        <span class="fu">sort</span> <span class="at">-V</span> <span class="at">-k1,1</span> <span class="at">-k2,2n</span>  <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb17-13"><a href="figure-3.html#cb17-13" tabindex="-1"></a>        <span class="fu">awk</span> <span class="st">&#39;{OFS=&quot;\t&quot;} {if ($4&lt;=0.05 &amp;&amp; $5&lt;0.1) {print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,&quot;Hypo&quot;} else if($4&lt;=0.05 &amp;&amp; $5&gt;0.1){print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,&quot;Hyper&quot;} else {print $0,&quot;NC&quot;}}&#39;</span> <span class="op">&gt;</span> metilene_res/<span class="va">${tag}</span>_de_novo_DMR.bed</span>
<span id="cb17-14"><a href="figure-3.html#cb17-14" tabindex="-1"></a><span class="co"># 2. tissue type MHB overlap with tissue type DMR </span></span>
<span id="cb17-15"><a href="figure-3.html#cb17-15" tabindex="-1"></a><span class="va">DMR_path</span><span class="op">=</span><span class="st">&quot;metilene_res/&quot;</span></span>
<span id="cb17-16"><a href="figure-3.html#cb17-16" tabindex="-1"></a><span class="va">tsMHB_path</span><span class="op">=</span><span class="st">&quot;Data/Processed_data/MHB/tumor&quot;</span></span>
<span id="cb17-17"><a href="figure-3.html#cb17-17" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="kw">`</span><span class="fu">ls</span> <span class="va">$DMR_path</span>/<span class="pp">*</span>_DMR.bed<span class="kw">`</span> </span>
<span id="cb17-18"><a href="figure-3.html#cb17-18" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb17-19"><a href="figure-3.html#cb17-19" tabindex="-1"></a>        <span class="va">a</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${i</span><span class="op">##</span><span class="pp">*</span>/<span class="va">}</span> <span class="kw">|</span><span class="fu">sed</span> <span class="st">&#39;s/_de_novo_DMR.bed//g&#39;</span><span class="va">)</span></span>
<span id="cb17-20"><a href="figure-3.html#cb17-20" tabindex="-1"></a>       <span class="cf">if</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;breast&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-21"><a href="figure-3.html#cb17-21" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_BRCA.bed&quot;</span></span>
<span id="cb17-22"><a href="figure-3.html#cb17-22" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;colon&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-23"><a href="figure-3.html#cb17-23" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_COAD.bed&quot;</span></span>
<span id="cb17-24"><a href="figure-3.html#cb17-24" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;esophagus&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-25"><a href="figure-3.html#cb17-25" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_ESCA.bed&quot;</span></span>
<span id="cb17-26"><a href="figure-3.html#cb17-26" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;head_and_neck&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-27"><a href="figure-3.html#cb17-27" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_HNSC.bed&quot;</span></span>
<span id="cb17-28"><a href="figure-3.html#cb17-28" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;lung&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-29"><a href="figure-3.html#cb17-29" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_NSCLC.bed&quot;</span></span>
<span id="cb17-30"><a href="figure-3.html#cb17-30" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;liver&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-31"><a href="figure-3.html#cb17-31" tabindex="-1"></a>                  <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_LIHC.bed&quot;</span></span>
<span id="cb17-32"><a href="figure-3.html#cb17-32" tabindex="-1"></a>          <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;pancreas&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-33"><a href="figure-3.html#cb17-33" tabindex="-1"></a>                    <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_PACA.bed&quot;</span></span>
<span id="cb17-34"><a href="figure-3.html#cb17-34" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;ovary&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-35"><a href="figure-3.html#cb17-35" tabindex="-1"></a>                    <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_OV.bed&quot;</span></span>
<span id="cb17-36"><a href="figure-3.html#cb17-36" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;stomach&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-37"><a href="figure-3.html#cb17-37" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_STAD.bed&quot;</span></span>
<span id="cb17-38"><a href="figure-3.html#cb17-38" tabindex="-1"></a>        <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">&quot;</span><span class="va">${a}</span><span class="st">&quot;</span> <span class="ot">==</span> <span class="pp">*</span><span class="st">&quot;thyroid&quot;</span><span class="pp">*</span> <span class="kw">]];</span><span class="cf">then</span></span>
<span id="cb17-39"><a href="figure-3.html#cb17-39" tabindex="-1"></a>                <span class="va">tag_T</span><span class="op">=</span><span class="st">&quot;MHB_THCA.bed&quot;</span></span>
<span id="cb17-40"><a href="figure-3.html#cb17-40" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb17-41"><a href="figure-3.html#cb17-41" tabindex="-1"></a>        <span class="va">tsMHB</span><span class="op">=</span><span class="va">${tsMHB_path}</span>/<span class="va">${tag_T}</span></span>
<span id="cb17-42"><a href="figure-3.html#cb17-42" tabindex="-1"></a>        <span class="ex">bedtools</span> intersect <span class="at">-a</span> <span class="va">${tsMHB}</span> <span class="at">-b</span>  <span class="va">${i}</span> <span class="at">-wa</span> <span class="at">-wb</span> <span class="at">-loj</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb17-43"><a href="figure-3.html#cb17-43" tabindex="-1"></a>        <span class="fu">awk</span> <span class="st">&#39;{OFS=&quot;\t&quot;} {if ($14==&quot;.&quot;) {print $1,$2,$3,&quot;&#39;&quot;</span><span class="va">${a}</span><span class="st">&quot;&#39;&quot;,&quot;NC&quot;} else {print $1,$2,$3,&quot;&#39;&quot;</span><span class="va">${a}</span><span class="st">&quot;&#39;&quot;,$14}}&#39;</span><span class="kw">|</span><span class="dt">\</span></span>
<span id="cb17-44"><a href="figure-3.html#cb17-44" tabindex="-1"></a>        <span class="ex">bedtools</span> merge <span class="at">-i</span> <span class="at">-</span> <span class="at">-c</span> 4,5,5 <span class="at">-o</span> distinct,distinct,count_distinct <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb17-45"><a href="figure-3.html#cb17-45" tabindex="-1"></a>        <span class="fu">sort</span> <span class="at">-k1,1</span> <span class="at">-k2,2n</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb17-46"><a href="figure-3.html#cb17-46" tabindex="-1"></a>        <span class="fu">awk</span> <span class="st">&#39;{OFS=&quot;\t&quot;} {if($6!=1) {print $1,$2,$3,$4,$5,$6,&quot;ND&quot;}else{print $0,$5}}&#39;</span> <span class="op">&gt;</span>tsMHB_DMR_Overlap/<span class="va">${a}</span>_MHB_DMR.txt</span>
<span id="cb17-47"><a href="figure-3.html#cb17-47" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="figure-3.html#cb18-1" tabindex="-1"></a><span class="co"># Cancer type MHB overlap Cancer type DMR</span></span>
<span id="cb18-2"><a href="figure-3.html#cb18-2" tabindex="-1"></a>files<span class="ot">=</span><span class="fu">list.files</span>(<span class="st">&quot;tsMHB_DMR_Overlap&quot;</span>)</span>
<span id="cb18-3"><a href="figure-3.html#cb18-3" tabindex="-1"></a><span class="co"># BRCA MHB</span></span>
<span id="cb18-4"><a href="figure-3.html#cb18-4" tabindex="-1"></a>BRCA_MHB_DMR <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;tsMHB_DMR_Overlap/breast_MHB_DMR.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>F)</span>
<span id="cb18-5"><a href="figure-3.html#cb18-5" tabindex="-1"></a>BRCA_count <span class="ot">=</span> BRCA_MHB_DMR <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(V4,V7) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">DMR_count=</span><span class="fu">n</span>())</span>
<span id="cb18-6"><a href="figure-3.html#cb18-6" tabindex="-1"></a>MHB_DMR_CS <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb18-7"><a href="figure-3.html#cb18-7" tabindex="-1"></a>MHB_DMR_CS[[<span class="st">&quot;breast&quot;</span>]] <span class="ot">&lt;-</span> BRCA_count</span>
<span id="cb18-8"><a href="figure-3.html#cb18-8" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> files[<span class="sc">-</span><span class="dv">1</span>]){</span>
<span id="cb18-9"><a href="figure-3.html#cb18-9" tabindex="-1"></a>    tag <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_MHB_DMR.txt&quot;</span>,<span class="st">&quot;&quot;</span>,i)</span>
<span id="cb18-10"><a href="figure-3.html#cb18-10" tabindex="-1"></a>    mx <span class="ot">=</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;tsMHB_DMR_Overlap/&quot;</span>,i),<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>F)</span>
<span id="cb18-11"><a href="figure-3.html#cb18-11" tabindex="-1"></a>    nx <span class="ot">=</span> mx <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(V4,V7) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">DMR_count=</span><span class="fu">n</span>())</span>
<span id="cb18-12"><a href="figure-3.html#cb18-12" tabindex="-1"></a>    MHB_DMR_CS[[tag]] <span class="ot">&lt;-</span>nx</span>
<span id="cb18-13"><a href="figure-3.html#cb18-13" tabindex="-1"></a>}</span>
<span id="cb18-14"><a href="figure-3.html#cb18-14" tabindex="-1"></a>MHB_DMR_frq <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind,MHB_DMR_CS))</span>
<span id="cb18-15"><a href="figure-3.html#cb18-15" tabindex="-1"></a>MHB_DMR_frq<span class="sc">$</span>type <span class="ot">=</span> <span class="fu">paste0</span>(MHB_DMR_frq<span class="sc">$</span>V4,<span class="st">&quot;_cancer&quot;</span>)</span>
<span id="cb18-16"><a href="figure-3.html#cb18-16" tabindex="-1"></a>MHB_DMR_frq <span class="ot">=</span> MHB_DMR_frq <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(type) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Freq=</span><span class="dv">100</span><span class="sc">*</span>DMR_count<span class="sc">/</span><span class="fu">sum</span>(DMR_count))</span>
<span id="cb18-17"><a href="figure-3.html#cb18-17" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">&quot;Data/Figures/Fig3/Fig3.A.pdf&quot;</span>,<span class="at">width=</span><span class="dv">4</span>,<span class="at">height=</span><span class="fl">5.5</span>)</span>
<span id="cb18-18"><a href="figure-3.html#cb18-18" tabindex="-1"></a>p<span class="ot">&lt;-</span><span class="fu">ggplot</span>(MHB_DMR_frq,<span class="fu">aes</span>(<span class="at">x=</span>type,<span class="at">y=</span>DMR_count)) <span class="sc">+</span></span>
<span id="cb18-19"><a href="figure-3.html#cb18-19" tabindex="-1"></a>          <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill=</span>V7),<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="at">position =</span> <span class="fu">position_fill</span>(<span class="at">reverse =</span> <span class="cn">TRUE</span>))<span class="sc">+</span></span>
<span id="cb18-20"><a href="figure-3.html#cb18-20" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;&quot;</span>,<span class="at">y=</span><span class="st">&quot;Percent (%)&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-21"><a href="figure-3.html#cb18-21" tabindex="-1"></a>          <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Set1&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-22"><a href="figure-3.html#cb18-22" tabindex="-1"></a>          <span class="fu">scale_y_continuous</span>(<span class="at">breaks=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.25</span>,<span class="fl">0.5</span>,<span class="fl">0.75</span>,<span class="dv">1</span>),<span class="at">labels=</span><span class="dv">100</span><span class="sc">*</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.25</span>,<span class="fl">0.5</span>,<span class="fl">0.75</span>,<span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb18-23"><a href="figure-3.html#cb18-23" tabindex="-1"></a>          <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb18-24"><a href="figure-3.html#cb18-24" tabindex="-1"></a>          <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-25"><a href="figure-3.html#cb18-25" tabindex="-1"></a>                <span class="at">panel.grid =</span><span class="fu">element_blank</span>(),</span>
<span id="cb18-26"><a href="figure-3.html#cb18-26" tabindex="-1"></a>                <span class="at">axis.ticks.length=</span><span class="fu">unit</span>(<span class="dv">5</span>,<span class="st">&#39;pt&#39;</span>),</span>
<span id="cb18-27"><a href="figure-3.html#cb18-27" tabindex="-1"></a>                <span class="at">axis.text.x =</span><span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>,<span class="at">vjust=</span><span class="fl">0.5</span>),</span>
<span id="cb18-28"><a href="figure-3.html#cb18-28" tabindex="-1"></a>                <span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">hjust=</span><span class="fl">0.5</span>),</span>
<span id="cb18-29"><a href="figure-3.html#cb18-29" tabindex="-1"></a>            <span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-30"><a href="figure-3.html#cb18-30" tabindex="-1"></a>          <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb18-31"><a href="figure-3.html#cb18-31" tabindex="-1"></a><span class="fu">print</span>(p)</span>
<span id="cb18-32"><a href="figure-3.html#cb18-32" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p><img src="Data/Figures/Fig3/Fig3.A.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="b-c.-enrichment-of-cancer-mhbs-in-dmr-lola" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> (B-C). Enrichment of Cancer MHBs in DMR (LOLA)<a href="figure-3.html#b-c.-enrichment-of-cancer-mhbs-in-dmr-lola" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="figure-3.html#cb19-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;LOLA&quot;</span>)</span>
<span id="cb19-2"><a href="figure-3.html#cb19-2" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb19-3"><a href="figure-3.html#cb19-3" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb19-4"><a href="figure-3.html#cb19-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb19-5"><a href="figure-3.html#cb19-5" tabindex="-1"></a>mhb_clusters<span class="ot">=</span><span class="st">&quot;Data/Processed_data/MHB/mhb_clusters/&quot;</span></span>
<span id="cb19-6"><a href="figure-3.html#cb19-6" tabindex="-1"></a>background<span class="ot">=</span><span class="st">&quot;Data/Processed_data/MHB/MHB_11_cancers_clusters.bed&quot;</span></span>
<span id="cb19-7"><a href="figure-3.html#cb19-7" tabindex="-1"></a><span class="do">## Hyper DMR</span></span>
<span id="cb19-8"><a href="figure-3.html#cb19-8" tabindex="-1"></a>files <span class="ot">=</span> <span class="fu">list.files</span>(mhb_clusters)</span>
<span id="cb19-9"><a href="figure-3.html#cb19-9" tabindex="-1"></a>locResults <span class="ot">=</span> <span class="fu">lapply</span>(files,<span class="cf">function</span>(i){</span>
<span id="cb19-10"><a href="figure-3.html#cb19-10" tabindex="-1"></a>  mhb<span class="ot">=</span><span class="fu">readBed</span>(<span class="fu">paste0</span>(mhb_clusters,i))</span>
<span id="cb19-11"><a href="figure-3.html#cb19-11" tabindex="-1"></a>  <span class="co"># Universe_Sets Background</span></span>
<span id="cb19-12"><a href="figure-3.html#cb19-12" tabindex="-1"></a>    universe_set<span class="ot">=</span><span class="fu">readBed</span>(background)</span>
<span id="cb19-13"><a href="figure-3.html#cb19-13" tabindex="-1"></a>  <span class="co"># Load regionDB</span></span>
<span id="cb19-14"><a href="figure-3.html#cb19-14" tabindex="-1"></a>    states<span class="ot">=</span><span class="fu">loadRegionDB</span>(<span class="st">&quot;Data/Processed_data/MR/LOLA_Hyper_DB/&quot;</span>)</span>
<span id="cb19-15"><a href="figure-3.html#cb19-15" tabindex="-1"></a>  <span class="co"># Runing LOLA</span></span>
<span id="cb19-16"><a href="figure-3.html#cb19-16" tabindex="-1"></a>    <span class="fu">runLOLA</span>(mhb, universe_set, states, <span class="at">cores=</span><span class="dv">10</span>)</span>
<span id="cb19-17"><a href="figure-3.html#cb19-17" tabindex="-1"></a>})</span>
<span id="cb19-18"><a href="figure-3.html#cb19-18" tabindex="-1"></a><span class="fu">names</span>(locResults) <span class="ot">&lt;-</span> files</span>
<span id="cb19-19"><a href="figure-3.html#cb19-19" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(files,<span class="cf">function</span>(i){</span>
<span id="cb19-20"><a href="figure-3.html#cb19-20" tabindex="-1"></a>  mx <span class="ot">=</span> locResults[[i]][,<span class="fu">c</span>(<span class="st">&quot;pValueLog&quot;</span>,<span class="st">&quot;filename&quot;</span>)] <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Type=</span>i,<span class="at">q=</span><span class="fu">p.adjust</span>(<span class="at">p=</span><span class="dv">10</span><span class="sc">^</span>(<span class="sc">-</span>pValueLog),<span class="at">method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">n=</span><span class="dv">10</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-21"><a href="figure-3.html#cb19-21" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pValueLog)</span>
<span id="cb19-22"><a href="figure-3.html#cb19-22" tabindex="-1"></a>})</span>
<span id="cb19-23"><a href="figure-3.html#cb19-23" tabindex="-1"></a>data <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind,data))</span>
<span id="cb19-24"><a href="figure-3.html#cb19-24" tabindex="-1"></a>dta <span class="ot">=</span> <span class="fu">dcast</span>(data,Type<span class="sc">~</span>filename)</span>
<span id="cb19-25"><a href="figure-3.html#cb19-25" tabindex="-1"></a><span class="fu">rownames</span>(dta) <span class="ot">=</span> dta<span class="sc">$</span>Type;dta<span class="sc">$</span>Type<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb19-26"><a href="figure-3.html#cb19-26" tabindex="-1"></a>dta <span class="ot">=</span> dta[<span class="fu">c</span>(<span class="st">&quot;BRCA_cluster.bed&quot;</span>,<span class="st">&quot;CESC_cluster.bed&quot;</span>,<span class="st">&quot;COAD_cluster.bed&quot;</span>,<span class="st">&quot;ESCA_cluster.bed&quot;</span>,<span class="st">&quot;HNSC_cluster.bed&quot;</span>,<span class="st">&quot;LIHC_cluster.bed&quot;</span>,<span class="st">&quot;NSCLC_cluster.bed&quot;</span>,<span class="st">&quot;OV_cluster.bed&quot;</span>,<span class="st">&quot;PACA_cluster.bed&quot;</span>,<span class="st">&quot;STAD_cluster.bed&quot;</span>, <span class="st">&quot;THCA_cluster.bed&quot;</span>,<span class="st">&quot;cluster12.bed&quot;</span>,<span class="st">&quot;cluster13.bed&quot;</span>,<span class="st">&quot;cluster14.bed&quot;</span>,<span class="st">&quot;cluster15.bed&quot;</span>,<span class="st">&quot;cluster16.bed&quot;</span>),]</span>
<span id="cb19-27"><a href="figure-3.html#cb19-27" tabindex="-1"></a>dta_t <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(dta)</span>
<span id="cb19-28"><a href="figure-3.html#cb19-28" tabindex="-1"></a><span class="co"># filter FDR &lt; 0.01  and then ranking</span></span>
<span id="cb19-29"><a href="figure-3.html#cb19-29" tabindex="-1"></a>dta_t[dta_t<span class="sc">&lt;</span><span class="dv">2</span>]<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb19-30"><a href="figure-3.html#cb19-30" tabindex="-1"></a>rank_t <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb19-31"><a href="figure-3.html#cb19-31" tabindex="-1"></a>    m <span class="ot">=</span> <span class="fu">rank</span>(<span class="fu">array</span>(<span class="sc">-</span>x),<span class="at">na.last=</span><span class="st">&quot;keep&quot;</span>,<span class="at">ties.method =</span> <span class="st">&quot;min&quot;</span>)</span>
<span id="cb19-32"><a href="figure-3.html#cb19-32" tabindex="-1"></a>    <span class="fu">return</span>(m)</span>
<span id="cb19-33"><a href="figure-3.html#cb19-33" tabindex="-1"></a>}</span>
<span id="cb19-34"><a href="figure-3.html#cb19-34" tabindex="-1"></a>data_rank<span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(dta_t,<span class="dv">1</span>,rank_t)))</span>
<span id="cb19-35"><a href="figure-3.html#cb19-35" tabindex="-1"></a><span class="fu">names</span>(data_rank) <span class="ot">=</span> <span class="fu">names</span>(dta_t)</span>
<span id="cb19-36"><a href="figure-3.html#cb19-36" tabindex="-1"></a><span class="co"># PLOT</span></span>
<span id="cb19-37"><a href="figure-3.html#cb19-37" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">&quot;Data/Figures/Fig3/Fig3.B.pdf&quot;</span>,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height=</span><span class="dv">6</span>)</span>
<span id="cb19-38"><a href="figure-3.html#cb19-38" tabindex="-1"></a>p<span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(data_rank,<span class="at">main=</span><span class="st">&quot;MHBs Overlap with Hyper-DMRs&quot;</span>,</span>
<span id="cb19-39"><a href="figure-3.html#cb19-39" tabindex="-1"></a>             <span class="at">show_colnames=</span>T,<span class="at">show_rownames=</span>T,</span>
<span id="cb19-40"><a href="figure-3.html#cb19-40" tabindex="-1"></a>             <span class="at">cluster_rows=</span>F,<span class="at">cluster_cols=</span>F,</span>
<span id="cb19-41"><a href="figure-3.html#cb19-41" tabindex="-1"></a>             <span class="at">scale=</span><span class="st">&quot;none&quot;</span>,<span class="at">angle_col=</span><span class="dv">45</span>,</span>
<span id="cb19-42"><a href="figure-3.html#cb19-42" tabindex="-1"></a>             <span class="at">color=</span><span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;firebrick&quot;</span>,<span class="st">&quot;white&quot;</span>,<span class="st">&quot;darkblue&quot;</span>))(<span class="dv">1000</span>))</span>
<span id="cb19-43"><a href="figure-3.html#cb19-43" tabindex="-1"></a><span class="fu">print</span>(p)</span>
<span id="cb19-44"><a href="figure-3.html#cb19-44" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb19-45"><a href="figure-3.html#cb19-45" tabindex="-1"></a><span class="do">## Hypo DMR</span></span>
<span id="cb19-46"><a href="figure-3.html#cb19-46" tabindex="-1"></a>locResults <span class="ot">=</span> <span class="fu">lapply</span>(files,<span class="cf">function</span>(i){</span>
<span id="cb19-47"><a href="figure-3.html#cb19-47" tabindex="-1"></a>  mhb<span class="ot">=</span><span class="fu">readBed</span>(<span class="fu">paste0</span>(mhb_clusters,i))</span>
<span id="cb19-48"><a href="figure-3.html#cb19-48" tabindex="-1"></a>  <span class="co"># Universe_Sets Background</span></span>
<span id="cb19-49"><a href="figure-3.html#cb19-49" tabindex="-1"></a>    universe_set<span class="ot">=</span><span class="fu">readBed</span>(background)</span>
<span id="cb19-50"><a href="figure-3.html#cb19-50" tabindex="-1"></a>  <span class="co"># Load regionDB</span></span>
<span id="cb19-51"><a href="figure-3.html#cb19-51" tabindex="-1"></a>    states<span class="ot">=</span><span class="fu">loadRegionDB</span>(<span class="st">&quot;Data/Processed_data/MR/LOLA_Hypo_DB/&quot;</span>)</span>
<span id="cb19-52"><a href="figure-3.html#cb19-52" tabindex="-1"></a>  <span class="co"># Runing LOLA</span></span>
<span id="cb19-53"><a href="figure-3.html#cb19-53" tabindex="-1"></a>    <span class="fu">runLOLA</span>(mhb, universe_set, states, <span class="at">cores=</span><span class="dv">10</span>)</span>
<span id="cb19-54"><a href="figure-3.html#cb19-54" tabindex="-1"></a>})</span>
<span id="cb19-55"><a href="figure-3.html#cb19-55" tabindex="-1"></a><span class="fu">names</span>(locResults) <span class="ot">&lt;-</span> files</span>
<span id="cb19-56"><a href="figure-3.html#cb19-56" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(files,<span class="cf">function</span>(i){</span>
<span id="cb19-57"><a href="figure-3.html#cb19-57" tabindex="-1"></a>  mx <span class="ot">=</span> locResults[[i]][,<span class="fu">c</span>(<span class="st">&quot;pValueLog&quot;</span>,<span class="st">&quot;filename&quot;</span>)] <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Type=</span>i,<span class="at">q=</span><span class="fu">p.adjust</span>(<span class="at">p=</span><span class="dv">10</span><span class="sc">^</span>(<span class="sc">-</span>pValueLog),<span class="at">method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">n=</span><span class="dv">10</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-58"><a href="figure-3.html#cb19-58" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pValueLog)</span>
<span id="cb19-59"><a href="figure-3.html#cb19-59" tabindex="-1"></a>})</span>
<span id="cb19-60"><a href="figure-3.html#cb19-60" tabindex="-1"></a>data <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind,data))</span>
<span id="cb19-61"><a href="figure-3.html#cb19-61" tabindex="-1"></a>dta <span class="ot">=</span> <span class="fu">dcast</span>(data,Type<span class="sc">~</span>filename)</span>
<span id="cb19-62"><a href="figure-3.html#cb19-62" tabindex="-1"></a><span class="fu">rownames</span>(dta) <span class="ot">=</span> dta<span class="sc">$</span>Type;dta<span class="sc">$</span>Type<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb19-63"><a href="figure-3.html#cb19-63" tabindex="-1"></a>dta <span class="ot">=</span> dta[<span class="fu">c</span>(<span class="st">&quot;BRCA_cluster.bed&quot;</span>,<span class="st">&quot;CESC_cluster.bed&quot;</span>,<span class="st">&quot;COAD_cluster.bed&quot;</span>,<span class="st">&quot;ESCA_cluster.bed&quot;</span>,<span class="st">&quot;HNSC_cluster.bed&quot;</span>,<span class="st">&quot;LIHC_cluster.bed&quot;</span>,<span class="st">&quot;NSCLC_cluster.bed&quot;</span>,<span class="st">&quot;OV_cluster.bed&quot;</span>,<span class="st">&quot;PACA_cluster.bed&quot;</span>,<span class="st">&quot;STAD_cluster.bed&quot;</span>, <span class="st">&quot;THCA_cluster.bed&quot;</span>,<span class="st">&quot;cluster12.bed&quot;</span>,<span class="st">&quot;cluster13.bed&quot;</span>,<span class="st">&quot;cluster14.bed&quot;</span>,<span class="st">&quot;cluster15.bed&quot;</span>,<span class="st">&quot;cluster16.bed&quot;</span>),]</span>
<span id="cb19-64"><a href="figure-3.html#cb19-64" tabindex="-1"></a>dta_t <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(dta)</span>
<span id="cb19-65"><a href="figure-3.html#cb19-65" tabindex="-1"></a><span class="co"># filter FDR &lt; 0.01 and then ranking </span></span>
<span id="cb19-66"><a href="figure-3.html#cb19-66" tabindex="-1"></a>dta_t[dta_t<span class="sc">&lt;</span><span class="dv">2</span>]<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb19-67"><a href="figure-3.html#cb19-67" tabindex="-1"></a>rank_t <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb19-68"><a href="figure-3.html#cb19-68" tabindex="-1"></a>    m <span class="ot">=</span> <span class="fu">rank</span>(<span class="fu">array</span>(<span class="sc">-</span>x),<span class="at">na.last=</span><span class="st">&quot;keep&quot;</span>,<span class="at">ties.method =</span> <span class="st">&quot;min&quot;</span>)</span>
<span id="cb19-69"><a href="figure-3.html#cb19-69" tabindex="-1"></a>    <span class="fu">return</span>(m)</span>
<span id="cb19-70"><a href="figure-3.html#cb19-70" tabindex="-1"></a>}</span>
<span id="cb19-71"><a href="figure-3.html#cb19-71" tabindex="-1"></a>data_rank<span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(dta_t,<span class="dv">1</span>,rank_t)))</span>
<span id="cb19-72"><a href="figure-3.html#cb19-72" tabindex="-1"></a><span class="fu">names</span>(data_rank) <span class="ot">=</span> <span class="fu">names</span>(dta_t)</span>
<span id="cb19-73"><a href="figure-3.html#cb19-73" tabindex="-1"></a><span class="co"># PLOT</span></span>
<span id="cb19-74"><a href="figure-3.html#cb19-74" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">&quot;Data/Figures/Fig3/Fig3.C.pdf&quot;</span>,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height=</span><span class="dv">6</span>)</span>
<span id="cb19-75"><a href="figure-3.html#cb19-75" tabindex="-1"></a>p<span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(data_rank,<span class="at">main=</span><span class="st">&quot;MHBs Overlap with Hypo-DMRs&quot;</span>,</span>
<span id="cb19-76"><a href="figure-3.html#cb19-76" tabindex="-1"></a>             <span class="at">show_colnames=</span>T,<span class="at">show_rownames=</span>T,</span>
<span id="cb19-77"><a href="figure-3.html#cb19-77" tabindex="-1"></a>             <span class="at">cluster_rows=</span>F,<span class="at">cluster_cols=</span>F,</span>
<span id="cb19-78"><a href="figure-3.html#cb19-78" tabindex="-1"></a>             <span class="at">scale=</span><span class="st">&quot;none&quot;</span>,<span class="at">angle_col=</span><span class="dv">45</span>,</span>
<span id="cb19-79"><a href="figure-3.html#cb19-79" tabindex="-1"></a>             <span class="at">color=</span><span class="fu">colorRampPalette</span>(<span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;firebrick&quot;</span>,<span class="st">&quot;white&quot;</span>,<span class="st">&quot;darkblue&quot;</span>))(<span class="dv">1000</span>))</span>
<span id="cb19-80"><a href="figure-3.html#cb19-80" tabindex="-1"></a><span class="fu">print</span>(p)</span>
<span id="cb19-81"><a href="figure-3.html#cb19-81" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-35"></span>
<img src="Data/Figures/Fig3/Fig3.B.png" alt="Fig3.B-C" width="40%" id="fig-unnamed-chunk-35-1" /><img src="Data/Figures/Fig3/Fig3.C.png" alt="Fig3.B-C" width="40%" id="fig-unnamed-chunk-35-2" />
<p class="caption">
 3.1: Fig3.B-C
</p>
</div>
</div>
<div id="d-e.-escc-dnam-and-rna-profiles-gse149608" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> (D-E). ESCC DNAm and RNA profiles (GSE149608)<a href="figure-3.html#d-e.-escc-dnam-and-rna-profiles-gse149608" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="figure-3.html#cb20-1" tabindex="-1"></a><span class="co"># Association of MHBs and dysregulation of gene expression in ESCC</span></span>
<span id="cb20-2"><a href="figure-3.html#cb20-2" tabindex="-1"></a><span class="fu">library</span>(rGREAT)</span>
<span id="cb20-3"><a href="figure-3.html#cb20-3" tabindex="-1"></a><span class="fu">library</span>(regioneR)</span>
<span id="cb20-4"><a href="figure-3.html#cb20-4" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb20-5"><a href="figure-3.html#cb20-5" tabindex="-1"></a><span class="co"># GREAT genome genes (hg19)</span></span>
<span id="cb20-6"><a href="figure-3.html#cb20-6" tabindex="-1"></a>Tss_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;Data/Processed_data/GREATv4.genes.promoter_UD1000.hg19.bed&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="figure-3.html#cb20-7" tabindex="-1"></a>            <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;start&quot;</span>,<span class="st">&quot;end&quot;</span>,<span class="st">&quot;strand&quot;</span>,<span class="st">&quot;genes&quot;</span>))</span>
<span id="cb20-8"><a href="figure-3.html#cb20-8" tabindex="-1"></a><span class="co"># Step1: MHB</span></span>
<span id="cb20-9"><a href="figure-3.html#cb20-9" tabindex="-1"></a>ESCC_T_MHB <span class="ot">&lt;-</span> <span class="st">&quot;Data/Processed_data/MHB/Public/MHB_ESCC.bed&quot;</span></span>
<span id="cb20-10"><a href="figure-3.html#cb20-10" tabindex="-1"></a>ESCC_N_MHB <span class="ot">&lt;-</span> <span class="st">&quot;Data/Processed_data/MHB/Public/MHB_Esophagus_N.bed&quot;</span></span>
<span id="cb20-11"><a href="figure-3.html#cb20-11" tabindex="-1"></a><span class="co"># MHB gene annotation</span></span>
<span id="cb20-12"><a href="figure-3.html#cb20-12" tabindex="-1"></a>MHBGenes_TSS <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(ESCC_T_MHB,ESCC_N_MHB),<span class="cf">function</span>(y){</span>
<span id="cb20-13"><a href="figure-3.html#cb20-13" tabindex="-1"></a>                    <span class="fu">toGRanges</span>(y) <span class="sc">%&gt;%</span> <span class="fu">great</span>(<span class="st">&quot;MSigDB:H&quot;</span>,<span class="st">&quot;GREAT:hg19&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">getRegionGeneAssociations</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-14"><a href="figure-3.html#cb20-14" tabindex="-1"></a>                    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span>  <span class="fu">mutate</span>(<span class="at">annotated_genes=</span><span class="fu">lapply</span>(annotated_genes,<span class="cf">function</span>(x){</span>
<span id="cb20-15"><a href="figure-3.html#cb20-15" tabindex="-1"></a>                                                x <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb20-16"><a href="figure-3.html#cb20-16" tabindex="-1"></a>                                             }) <span class="sc">%&gt;%</span> <span class="fu">do.call</span>(rbind,.) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>(),</span>
<span id="cb20-17"><a href="figure-3.html#cb20-17" tabindex="-1"></a>                                             <span class="at">dist_to_TSS =</span> <span class="fu">lapply</span>(dist_to_TSS,<span class="cf">function</span>(x){</span>
<span id="cb20-18"><a href="figure-3.html#cb20-18" tabindex="-1"></a>                                                x <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb20-19"><a href="figure-3.html#cb20-19" tabindex="-1"></a>                                             }) <span class="sc">%&gt;%</span> <span class="fu">do.call</span>(rbind,.) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb20-20"><a href="figure-3.html#cb20-20" tabindex="-1"></a>              <span class="fu">separate_rows</span>(annotated_genes,dist_to_TSS,<span class="at">sep=</span><span class="st">&quot;,&quot;</span>,<span class="at">convert=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-21"><a href="figure-3.html#cb20-21" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">Type=</span><span class="fu">ifelse</span>(<span class="fu">str_detect</span>(y,<span class="st">&quot;ESCC&quot;</span>),<span class="st">&quot;esophagus_T&quot;</span>,<span class="st">&quot;esophagus_N&quot;</span>))})</span>
<span id="cb20-22"><a href="figure-3.html#cb20-22" tabindex="-1"></a><span class="fu">names</span>(MHBGenes_TSS) <span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;esophagus_T&quot;</span>,<span class="st">&quot;esophagus_N&quot;</span>)</span>
<span id="cb20-23"><a href="figure-3.html#cb20-23" tabindex="-1"></a><span class="co"># step2: Load DEGs</span></span>
<span id="cb20-24"><a href="figure-3.html#cb20-24" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Data/Processed_data/ESCC/ESCC_log2TPM_Wilcox.RData&quot;</span>)</span>
<span id="cb20-25"><a href="figure-3.html#cb20-25" tabindex="-1"></a>ESCC_DEGs <span class="ot">&lt;-</span> ESCC_log2TPM_Wilcox </span>
<span id="cb20-26"><a href="figure-3.html#cb20-26" tabindex="-1"></a>DEGs_up <span class="ot">&lt;-</span>  ESCC_DEGs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(P_DEG<span class="sc">==</span><span class="st">&quot;Up&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb20-27"><a href="figure-3.html#cb20-27" tabindex="-1"></a>DEGs_down <span class="ot">&lt;-</span>  ESCC_DEGs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(P_DEG<span class="sc">==</span><span class="st">&quot;Down&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb20-28"><a href="figure-3.html#cb20-28" tabindex="-1"></a><span class="co"># Step 3: Load DMR  (Wilcoxon or Metilene)</span></span>
<span id="cb20-29"><a href="figure-3.html#cb20-29" tabindex="-1"></a><span class="co"># 3.1 ESCC DNAme promoter +/-  TSS 1K (Differentially of Methylation of Promoters)</span></span>
<span id="cb20-30"><a href="figure-3.html#cb20-30" tabindex="-1"></a>DNAme_path<span class="ot">=</span><span class="st">&quot;DNAme/Sample_levels&quot;</span></span>
<span id="cb20-31"><a href="figure-3.html#cb20-31" tabindex="-1"></a>files <span class="ot">=</span> <span class="fu">list.files</span>(DNAme_path,<span class="at">pattern=</span><span class="st">&quot;_TSS1k.txt&quot;</span>)</span>
<span id="cb20-32"><a href="figure-3.html#cb20-32" tabindex="-1"></a>files <span class="ot">=</span> files[<span class="fu">str_detect</span>(files,<span class="st">&quot;esophagus&quot;</span>)]</span>
<span id="cb20-33"><a href="figure-3.html#cb20-33" tabindex="-1"></a></span>
<span id="cb20-34"><a href="figure-3.html#cb20-34" tabindex="-1"></a>ESCC_TSS_DNAme <span class="ot">&lt;-</span> <span class="fu">lapply</span>(files,<span class="cf">function</span>(x) {</span>
<span id="cb20-35"><a href="figure-3.html#cb20-35" tabindex="-1"></a>        <span class="fu">fread</span>(<span class="fu">paste0</span>(DNAme_path,<span class="st">&quot;/&quot;</span>,x))  <span class="sc">%&gt;%</span> </span>
<span id="cb20-36"><a href="figure-3.html#cb20-36" tabindex="-1"></a>        <span class="fu">transmute</span>(<span class="at">chr=</span>Chr,<span class="at">start=</span>Start <span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb20-37"><a href="figure-3.html#cb20-37" tabindex="-1"></a>                  <span class="at">end=</span>End,<span class="at">beta=</span><span class="fu">ifelse</span>(tBase<span class="sc">&gt;=</span><span class="dv">10</span>,<span class="fu">round</span>(mBase<span class="sc">/</span>tBase,<span class="dv">3</span>),<span class="cn">NA</span>),</span>
<span id="cb20-38"><a href="figure-3.html#cb20-38" tabindex="-1"></a>                  <span class="at">Type=</span><span class="fu">str_replace_all</span>(x,<span class="st">&quot;_TSS1k.txt&quot;</span>,<span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb20-39"><a href="figure-3.html#cb20-39" tabindex="-1"></a>            <span class="fu">cbind</span>(Tss_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">&quot;genes&quot;</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(genes,beta,Type) </span>
<span id="cb20-40"><a href="figure-3.html#cb20-40" tabindex="-1"></a>    })</span>
<span id="cb20-41"><a href="figure-3.html#cb20-41" tabindex="-1"></a>ESCC_DNAme <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,ESCC_TSS_DNAme) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()  <span class="sc">%&gt;%</span> </span>
<span id="cb20-42"><a href="figure-3.html#cb20-42" tabindex="-1"></a>            <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>Type,<span class="at">values_from=</span>beta,<span class="at">values_fn=</span>mean) <span class="sc">%&gt;%</span> </span>
<span id="cb20-43"><a href="figure-3.html#cb20-43" tabindex="-1"></a>            <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>)  <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb20-44"><a href="figure-3.html#cb20-44" tabindex="-1"></a><span class="co"># Wilcoxon test between T vs N</span></span>
<span id="cb20-45"><a href="figure-3.html#cb20-45" tabindex="-1"></a>Diff_DNAme <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb20-46"><a href="figure-3.html#cb20-46" tabindex="-1"></a>        pData <span class="ot">=</span> <span class="fu">names</span>(x)</span>
<span id="cb20-47"><a href="figure-3.html#cb20-47" tabindex="-1"></a>        Normal <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(<span class="fu">na.omit</span>(x[<span class="fu">grep</span>(<span class="st">&quot;N&quot;</span>,pData,<span class="at">value=</span>T)])))</span>
<span id="cb20-48"><a href="figure-3.html#cb20-48" tabindex="-1"></a>        Tumor <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(<span class="fu">na.omit</span>(x[<span class="fu">grep</span>(<span class="st">&quot;T&quot;</span>,pData,<span class="at">value=</span>T)])))</span>
<span id="cb20-49"><a href="figure-3.html#cb20-49" tabindex="-1"></a>        delta <span class="ot">=</span> <span class="fu">mean</span>(Tumor) <span class="sc">-</span> <span class="fu">mean</span>(Normal)</span>
<span id="cb20-50"><a href="figure-3.html#cb20-50" tabindex="-1"></a>       N <span class="ot">&lt;-</span> <span class="fu">mean</span>(Normal) ; T <span class="ot">&lt;-</span> <span class="fu">mean</span>(Tumor)</span>
<span id="cb20-51"><a href="figure-3.html#cb20-51" tabindex="-1"></a>        p_value <span class="ot">=</span><span class="fu">wilcox.test</span>(Normal,Tumor,<span class="at">paired=</span>T)<span class="sc">$</span>p.value</span>
<span id="cb20-52"><a href="figure-3.html#cb20-52" tabindex="-1"></a>        res <span class="ot">=</span> <span class="fu">c</span>(delta,N,T,p_value)</span>
<span id="cb20-53"><a href="figure-3.html#cb20-53" tabindex="-1"></a>        <span class="fu">return</span>(res)</span>
<span id="cb20-54"><a href="figure-3.html#cb20-54" tabindex="-1"></a>  }</span>
<span id="cb20-55"><a href="figure-3.html#cb20-55" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(ESCC_DNAme,<span class="dv">1</span>,Diff_DNAme)))  </span>
<span id="cb20-56"><a href="figure-3.html#cb20-56" tabindex="-1"></a><span class="fu">names</span>(res) <span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;delta&quot;</span>,<span class="st">&quot;Mean_N&quot;</span>,<span class="st">&quot;Mean_T&quot;</span>,<span class="st">&quot;P_value&quot;</span>) </span>
<span id="cb20-57"><a href="figure-3.html#cb20-57" tabindex="-1"></a>ESCC_DNAme_res <span class="ot">=</span> <span class="fu">cbind</span>(ESCC_DNAme,res) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">FDR =</span> <span class="fu">p.adjust</span>(P_value,<span class="at">method=</span><span class="st">&quot;BH&quot;</span>,<span class="at">n=</span><span class="fu">length</span>(P_value)),</span>
<span id="cb20-58"><a href="figure-3.html#cb20-58" tabindex="-1"></a>                    <span class="at">DM =</span> <span class="fu">ifelse</span>(delta<span class="sc">&gt;</span><span class="fl">0.1</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;=</span> <span class="fl">0.05</span>,<span class="st">&quot;Hyper&quot;</span>,</span>
<span id="cb20-59"><a href="figure-3.html#cb20-59" tabindex="-1"></a>                                <span class="fu">ifelse</span>(delta<span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.1</span> <span class="sc">&amp;</span> FDR <span class="sc">&lt;=</span> <span class="fl">0.05</span>,<span class="st">&quot;Hypo&quot;</span>,<span class="st">&quot;NC&quot;</span>)),</span>
<span id="cb20-60"><a href="figure-3.html#cb20-60" tabindex="-1"></a>                    <span class="at">T_beta =</span> <span class="fu">ifelse</span>(Mean_T<span class="sc">&gt;=</span><span class="fl">0.8</span>,<span class="st">&quot;High&quot;</span>,<span class="fu">ifelse</span>(Mean_T<span class="sc">&lt;=</span><span class="fl">0.2</span>,<span class="st">&quot;Low&quot;</span>,<span class="st">&quot;Intermediate&quot;</span>))) </span>
<span id="cb20-61"><a href="figure-3.html#cb20-61" tabindex="-1"></a><span class="co"># 3.2 Call DMR metilene </span></span>
<span id="cb20-62"><a href="figure-3.html#cb20-62" tabindex="-1"></a>DMR_Metilene <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;Data/Processed_data/ESCC/ESCC_T_N_de_novo_DMR.bed&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-63"><a href="figure-3.html#cb20-63" tabindex="-1"></a>               <span class="fu">toGRanges</span>() <span class="sc">%&gt;%</span> <span class="fu">great</span>(<span class="st">&quot;MSigDB:H&quot;</span>,<span class="st">&quot;GREAT:hg19&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">getRegionGeneAssociations</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-64"><a href="figure-3.html#cb20-64" tabindex="-1"></a>               <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span>  <span class="fu">mutate</span>(<span class="at">annotated_genes=</span><span class="fu">lapply</span>(annotated_genes,<span class="cf">function</span>(x){</span>
<span id="cb20-65"><a href="figure-3.html#cb20-65" tabindex="-1"></a>                                                x <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb20-66"><a href="figure-3.html#cb20-66" tabindex="-1"></a>                                             }) <span class="sc">%&gt;%</span> <span class="fu">do.call</span>(rbind,.) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>(),</span>
<span id="cb20-67"><a href="figure-3.html#cb20-67" tabindex="-1"></a>                                             <span class="at">dist_to_TSS =</span> <span class="fu">lapply</span>(dist_to_TSS,<span class="cf">function</span>(x){</span>
<span id="cb20-68"><a href="figure-3.html#cb20-68" tabindex="-1"></a>                                                x <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb20-69"><a href="figure-3.html#cb20-69" tabindex="-1"></a>                                             }) <span class="sc">%&gt;%</span> <span class="fu">do.call</span>(rbind,.) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb20-70"><a href="figure-3.html#cb20-70" tabindex="-1"></a>                                             <span class="fu">separate_rows</span>(annotated_genes,dist_to_TSS,<span class="at">sep=</span><span class="st">&quot;,&quot;</span>,<span class="at">convert=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-71"><a href="figure-3.html#cb20-71" tabindex="-1"></a><span class="co"># 3.3 Non-DMR  ## the genes are not regulated by DMR</span></span>
<span id="cb20-72"><a href="figure-3.html#cb20-72" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(ESCC_log2TPM),Tss_data <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes))</span>
<span id="cb20-73"><a href="figure-3.html#cb20-73" tabindex="-1"></a><span class="co"># Non DMR genes</span></span>
<span id="cb20-74"><a href="figure-3.html#cb20-74" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">1</span>){   <span class="do">## from metilene DMR </span></span>
<span id="cb20-75"><a href="figure-3.html#cb20-75" tabindex="-1"></a>   DMR_genes <span class="ot">&lt;-</span> DMR_Metilene <span class="sc">%&gt;%</span> <span class="fu">filter</span>(V11 <span class="sc">!=</span> <span class="st">&quot;NC&quot;</span>) <span class="sc">%&gt;%</span>  <span class="fu">pull</span>(annotated_genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()  <span class="do">##</span></span>
<span id="cb20-76"><a href="figure-3.html#cb20-76" tabindex="-1"></a>   Non_DMR_genes <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(common_genes, DMR_genes) </span>
<span id="cb20-77"><a href="figure-3.html#cb20-77" tabindex="-1"></a>}<span class="cf">else</span> {. <span class="do">## from Wilcoxon test</span></span>
<span id="cb20-78"><a href="figure-3.html#cb20-78" tabindex="-1"></a>   Non_DMR_genes <span class="ot">&lt;-</span> ESCC_DNAme_res <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(DM <span class="sc">==</span> <span class="st">&quot;NC&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb20-79"><a href="figure-3.html#cb20-79" tabindex="-1"></a>   Non_DMR_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(Non_DMR_genes,common_genes)</span>
<span id="cb20-80"><a href="figure-3.html#cb20-80" tabindex="-1"></a>}</span>
<span id="cb20-81"><a href="figure-3.html#cb20-81" tabindex="-1"></a><span class="co"># Revome the promoter delta Beta &gt; 0.1</span></span>
<span id="cb20-82"><a href="figure-3.html#cb20-82" tabindex="-1"></a>genes_N <span class="ot">&lt;-</span> ESCC_DNAme_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">abs</span>(delta) <span class="sc">&lt;</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb20-83"><a href="figure-3.html#cb20-83" tabindex="-1"></a>Non_DMR_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(Non_DMR_genes,genes_N)  </span>
<span id="cb20-84"><a href="figure-3.html#cb20-84" tabindex="-1"></a><span class="co"># save</span></span>
<span id="cb20-85"><a href="figure-3.html#cb20-85" tabindex="-1"></a>Non_DMR_Genes_TSS <span class="ot">&lt;-</span> Tss_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genes <span class="sc">%in%</span> Non_DMR_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb20-86"><a href="figure-3.html#cb20-86" tabindex="-1"></a>                                  <span class="fu">transmute</span>(<span class="at">chr=</span>chr,<span class="at">start=</span>start<span class="sc">+</span><span class="dv">500</span>,</span>
<span id="cb20-87"><a href="figure-3.html#cb20-87" tabindex="-1"></a>                                         <span class="at">end=</span>end<span class="dv">-499</span>,<span class="at">gene=</span>genes)</span>
<span id="cb20-88"><a href="figure-3.html#cb20-88" tabindex="-1"></a><span class="fu">write.table</span>(Non_DMR_Genes_TSS,<span class="at">file=</span><span class="st">&quot;ESCC_Non_DMR_GENES_TSS_500.bed&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F,<span class="at">quote=</span>F,<span class="at">col.names=</span>F)</span>
<span id="cb20-89"><a href="figure-3.html#cb20-89" tabindex="-1"></a></span>
<span id="cb20-90"><a href="figure-3.html#cb20-90" tabindex="-1"></a><span class="co"># 4.Integration analysis of MHB, DMRs, DEGs </span></span>
<span id="cb20-91"><a href="figure-3.html#cb20-91" tabindex="-1"></a>DNAme_MHB_Exp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;esophagus_T&quot;</span>,<span class="st">&quot;esophagus_N&quot;</span>),<span class="cf">function</span>(x){</span>
<span id="cb20-92"><a href="figure-3.html#cb20-92" tabindex="-1"></a>          </span>
<span id="cb20-93"><a href="figure-3.html#cb20-93" tabindex="-1"></a>          mhbgenes <span class="ot">&lt;-</span> MHBGenes_TSS[[x]] <span class="sc">%&gt;%</span> <span class="fu">pull</span>(annotated_genes) <span class="sc">%&gt;%</span>  <span class="fu">unique</span>()</span>
<span id="cb20-94"><a href="figure-3.html#cb20-94" tabindex="-1"></a>          <span class="co"># 1932</span></span>
<span id="cb20-95"><a href="figure-3.html#cb20-95" tabindex="-1"></a>          DEGs_up <span class="ot">&lt;-</span> ESCC_DEGs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-96"><a href="figure-3.html#cb20-96" tabindex="-1"></a>                    <span class="fu">filter</span>(genes <span class="sc">%in%</span> common_genes,P_DEG <span class="sc">==</span> <span class="st">&quot;Up&quot;</span>)  <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb20-97"><a href="figure-3.html#cb20-97" tabindex="-1"></a>          <span class="co"># 837</span></span>
<span id="cb20-98"><a href="figure-3.html#cb20-98" tabindex="-1"></a>          DEGs_down <span class="ot">&lt;-</span> ESCC_DEGs <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-99"><a href="figure-3.html#cb20-99" tabindex="-1"></a>                    <span class="fu">filter</span>(genes <span class="sc">%in%</span> common_genes, P_DEG <span class="sc">==</span> <span class="st">&quot;Down&quot;</span>)  <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb20-100"><a href="figure-3.html#cb20-100" tabindex="-1"></a>          NonDMR_DEG_up <span class="ot">&lt;-</span> <span class="fu">intersect</span>(DEGs_up,Non_DMR_genes)</span>
<span id="cb20-101"><a href="figure-3.html#cb20-101" tabindex="-1"></a>          NonDMR_DEG_DN <span class="ot">&lt;-</span>  <span class="fu">intersect</span>(DEGs_down,Non_DMR_genes)</span>
<span id="cb20-102"><a href="figure-3.html#cb20-102" tabindex="-1"></a></span>
<span id="cb20-103"><a href="figure-3.html#cb20-103" tabindex="-1"></a>         <span class="co"># 2 x 2 table</span></span>
<span id="cb20-104"><a href="figure-3.html#cb20-104" tabindex="-1"></a>         a <span class="ot">&lt;-</span> <span class="fu">intersect</span>(mhbgenes,NonDMR_DEG_up) <span class="sc">%&gt;%</span> <span class="fu">length</span>()  <span class="co"># 22</span></span>
<span id="cb20-105"><a href="figure-3.html#cb20-105" tabindex="-1"></a>         b <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(NonDMR_DEG_up,mhbgenes) <span class="sc">%&gt;%</span> <span class="fu">length</span>()    <span class="co"># 106</span></span>
<span id="cb20-106"><a href="figure-3.html#cb20-106" tabindex="-1"></a></span>
<span id="cb20-107"><a href="figure-3.html#cb20-107" tabindex="-1"></a>         d <span class="ot">&lt;-</span> <span class="fu">intersect</span>(mhbgenes,NonDMR_DEG_DN) <span class="sc">%&gt;%</span> <span class="fu">length</span>()  <span class="co"># 0</span></span>
<span id="cb20-108"><a href="figure-3.html#cb20-108" tabindex="-1"></a>         e <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(NonDMR_DEG_DN,mhbgenes) <span class="sc">%&gt;%</span> <span class="fu">length</span>()    <span class="co"># 25</span></span>
<span id="cb20-109"><a href="figure-3.html#cb20-109" tabindex="-1"></a>         </span>
<span id="cb20-110"><a href="figure-3.html#cb20-110" tabindex="-1"></a>         <span class="fu">cat</span>(a,b,d,e,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-111"><a href="figure-3.html#cb20-111" tabindex="-1"></a>         <span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(d,e,a,b),<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">nrow=</span><span class="dv">2</span>))<span class="sc">%&gt;%</span> broom<span class="sc">::</span><span class="fu">tidy</span>()</span>
<span id="cb20-112"><a href="figure-3.html#cb20-112" tabindex="-1"></a>})</span>
<span id="cb20-113"><a href="figure-3.html#cb20-113" tabindex="-1"></a><span class="co"># show case the cancer mhb associated genes with up-regulation and the methylation levels of promoter without any changes.</span></span>
<span id="cb20-114"><a href="figure-3.html#cb20-114" tabindex="-1"></a>mhbgenes <span class="ot">&lt;-</span>  mhbgenes <span class="ot">&lt;-</span> MHBGenes_TSS[[<span class="st">&quot;esophagus_T&quot;</span>]] <span class="sc">%&gt;%</span> <span class="fu">pull</span>(annotated_genes) <span class="sc">%&gt;%</span>  <span class="fu">unique</span>()</span>
<span id="cb20-115"><a href="figure-3.html#cb20-115" tabindex="-1"></a>NonDMR_DEG_up <span class="ot">&lt;-</span> <span class="fu">intersect</span>(DEGs_up,Non_DMR_genes)</span>
<span id="cb20-116"><a href="figure-3.html#cb20-116" tabindex="-1"></a>hint <span class="ot">&lt;-</span> <span class="fu">intersect</span>(mhbgenes,NonDMR_DEG_up)</span>
<span id="cb20-117"><a href="figure-3.html#cb20-117" tabindex="-1"></a><span class="co"># sample heatmap preparation</span></span>
<span id="cb20-118"><a href="figure-3.html#cb20-118" tabindex="-1"></a>data.DNAme <span class="ot">&lt;-</span> ESCC_DNAme_res <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">contains</span>(<span class="st">&quot;esophagus&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-119"><a href="figure-3.html#cb20-119" tabindex="-1"></a>                                 <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-120"><a href="figure-3.html#cb20-120" tabindex="-1"></a>                                 <span class="fu">filter</span>(genes <span class="sc">%in%</span> hint)  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(genes) <span class="sc">%&gt;%</span></span>
<span id="cb20-121"><a href="figure-3.html#cb20-121" tabindex="-1"></a>                                  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-122"><a href="figure-3.html#cb20-122" tabindex="-1"></a>                                  <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">gsub</span>(<span class="st">&quot;esophagus_&quot;</span>,<span class="st">&quot;&quot;</span>,.x))</span>
<span id="cb20-123"><a href="figure-3.html#cb20-123" tabindex="-1"></a>data.RNA <span class="ot">&lt;-</span> ESCC_log2TPM_Wilcox <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">contains</span>(<span class="st">&quot;GSM&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-124"><a href="figure-3.html#cb20-124" tabindex="-1"></a>                                 <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">str_split_i</span>(.x,<span class="st">&quot;_&quot;</span>,<span class="at">i=</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-125"><a href="figure-3.html#cb20-125" tabindex="-1"></a>                                 <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-126"><a href="figure-3.html#cb20-126" tabindex="-1"></a>                                 <span class="fu">filter</span>(genes <span class="sc">%in%</span> hint)  <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(genes) <span class="sc">%&gt;%</span> </span>
<span id="cb20-127"><a href="figure-3.html#cb20-127" tabindex="-1"></a>                                 <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;genes&quot;</span>) </span>
<span id="cb20-128"><a href="figure-3.html#cb20-128" tabindex="-1"></a><span class="co"># PLOT DNA methylation &amp; Gene expression in MHB-associated up-regulated genes</span></span>
<span id="cb20-129"><a href="figure-3.html#cb20-129" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb20-130"><a href="figure-3.html#cb20-130" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb20-131"><a href="figure-3.html#cb20-131" tabindex="-1"></a><span class="co"># Annotation </span></span>
<span id="cb20-132"><a href="figure-3.html#cb20-132" tabindex="-1"></a>mhb_N <span class="ot">&lt;-</span> MHBGenes_TSS[[<span class="st">&quot;esophagus_N&quot;</span>]] <span class="sc">%&gt;%</span> <span class="fu">pull</span>(annotated_genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()   </span>
<span id="cb20-133"><a href="figure-3.html#cb20-133" tabindex="-1"></a>mhb_Anno <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>hint,<span class="at">Normal=</span> (hint <span class="sc">%in%</span> mhb_N),<span class="at">Tumor=</span><span class="st">&quot;TRUE&quot;</span>)</span>
<span id="cb20-134"><a href="figure-3.html#cb20-134" tabindex="-1"></a><span class="co"># DNAme</span></span>
<span id="cb20-135"><a href="figure-3.html#cb20-135" tabindex="-1"></a>column_ha_1 <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">Assay =</span> <span class="fu">rep</span>(<span class="st">&quot;WGBS&quot;</span>,<span class="dv">20</span>) , <span class="at">Type =</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>( <span class="fu">names</span>(data.DNAme),<span class="st">&quot;N&quot;</span>),<span class="st">&quot;Normal&quot;</span>,<span class="st">&quot;Tumor&quot;</span>) )</span>
<span id="cb20-136"><a href="figure-3.html#cb20-136" tabindex="-1"></a>row_ha_1 <span class="ot">=</span> <span class="fu">rowAnnotation</span>(<span class="at">MHB_T =</span> <span class="fu">rep</span>(<span class="dv">1</span>,<span class="dv">22</span>), <span class="at">MHB_N=</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>(mhb_Anno<span class="sc">$</span>Normal,<span class="st">&quot;TRUE&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb20-137"><a href="figure-3.html#cb20-137" tabindex="-1"></a>colfun11 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&#39;darkblue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;firebrick&#39;</span>))</span>
<span id="cb20-138"><a href="figure-3.html#cb20-138" tabindex="-1"></a><span class="co"># RNA</span></span>
<span id="cb20-139"><a href="figure-3.html#cb20-139" tabindex="-1"></a>column_ha_2 <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">Assay =</span> <span class="fu">rep</span>(<span class="st">&quot;RNA&quot;</span>,<span class="dv">20</span>) , <span class="at">Type =</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>( <span class="fu">names</span>(data.DNAme),<span class="st">&quot;N&quot;</span>),<span class="st">&quot;Normal&quot;</span>,<span class="st">&quot;Tumor&quot;</span>) )</span>
<span id="cb20-140"><a href="figure-3.html#cb20-140" tabindex="-1"></a>row_ha_2 <span class="ot">=</span> <span class="fu">rowAnnotation</span>(<span class="at">MHB_T =</span> <span class="fu">rep</span>(<span class="dv">1</span>,<span class="dv">22</span>), <span class="at">MHB_N=</span> <span class="fu">ifelse</span>(<span class="fu">str_detect</span>(mhb_Anno<span class="sc">$</span>Normal,<span class="st">&quot;TRUE&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb20-141"><a href="figure-3.html#cb20-141" tabindex="-1"></a>colfun2 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">0</span>, <span class="dv">3</span>), <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&#39;darkblue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;firebrick&#39;</span>))</span>
<span id="cb20-142"><a href="figure-3.html#cb20-142" tabindex="-1"></a>data.RNA.scale <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(<span class="fu">as.matrix</span>(data.RNA))))</span>
<span id="cb20-143"><a href="figure-3.html#cb20-143" tabindex="-1"></a></span>
<span id="cb20-144"><a href="figure-3.html#cb20-144" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="st">&quot;NonDMR_MHB_22_DNAme_DEGs.pdf&quot;</span>,<span class="at">width=</span><span class="dv">13</span>,<span class="at">heigh=</span><span class="dv">8</span>)</span>
<span id="cb20-145"><a href="figure-3.html#cb20-145" tabindex="-1"></a>p11 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(<span class="fu">as.matrix</span>(data.DNAme),<span class="at">name=</span><span class="st">&quot;Beta&quot;</span>,</span>
<span id="cb20-146"><a href="figure-3.html#cb20-146" tabindex="-1"></a>                <span class="at">col=</span>colfun11,<span class="at">cluster_columns=</span> F,</span>
<span id="cb20-147"><a href="figure-3.html#cb20-147" tabindex="-1"></a>                <span class="at">top_annotation =</span> column_ha_1, <span class="at">left_annotation =</span> row_ha_1)</span>
<span id="cb20-148"><a href="figure-3.html#cb20-148" tabindex="-1"></a>p21 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(data.RNA.scale,<span class="at">name=</span><span class="st">&quot;log2TPM(scaled)&quot;</span>,</span>
<span id="cb20-149"><a href="figure-3.html#cb20-149" tabindex="-1"></a>                <span class="at">col=</span>colfun2,<span class="at">cluster_columns=</span> F,</span>
<span id="cb20-150"><a href="figure-3.html#cb20-150" tabindex="-1"></a>                <span class="at">top_annotation =</span> column_ha_2)</span>
<span id="cb20-151"><a href="figure-3.html#cb20-151" tabindex="-1"></a>p<span class="ot">&lt;-</span> p11<span class="sc">+</span>p21</span>
<span id="cb20-152"><a href="figure-3.html#cb20-152" tabindex="-1"></a><span class="fu">print</span>(p)</span>
<span id="cb20-153"><a href="figure-3.html#cb20-153" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb21-1"><a href="figure-3.html#cb21-1" tabindex="-1"></a><span class="co"># Mean methyltion profiles of the genes promoter regions (Non DMR genes).</span></span>
<span id="cb21-2"><a href="figure-3.html#cb21-2" tabindex="-1"></a><span class="va">path</span><span class="op">=</span><span class="st">&quot;DNAme_Exp_N_T/DNAme&quot;</span></span>
<span id="cb21-3"><a href="figure-3.html#cb21-3" tabindex="-1"></a><span class="co"># 1069  genes with Non-DMR</span></span>
<span id="cb21-4"><a href="figure-3.html#cb21-4" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> esophagus_N_CpG_BED6.bed esophagus_T_CpG_BED6.bed</span>
<span id="cb21-5"><a href="figure-3.html#cb21-5" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb21-6"><a href="figure-3.html#cb21-6" tabindex="-1"></a>    <span class="ex">beta_profile_region.py</span> <span class="dt">\</span></span>
<span id="cb21-7"><a href="figure-3.html#cb21-7" tabindex="-1"></a>    <span class="at">-r</span> ESCC_Non_DMR_GENES_TSS_500.bed <span class="dt">\</span></span>
<span id="cb21-8"><a href="figure-3.html#cb21-8" tabindex="-1"></a>    <span class="at">-i</span> <span class="va">${path}</span>/<span class="va">${i}</span> <span class="dt">\</span></span>
<span id="cb21-9"><a href="figure-3.html#cb21-9" tabindex="-1"></a>    <span class="at">-o</span> profiles/<span class="va">${i</span><span class="op">%</span>_<span class="pp">*</span><span class="va">}</span>_NonDMR_CpGTools_TSS1K </span>
<span id="cb21-10"><a href="figure-3.html#cb21-10" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-38"></span>
<img src="Data/Figures/Fig3/Fig3.D.png" alt="Fig3.D-E" width="50%" id="fig-unnamed-chunk-38-1" /><img src="Data/Figures/Fig3/Fig3.E.png" alt="Fig3.D-E" width="50%" id="fig-unnamed-chunk-38-2" />
<p class="caption">
 3.2: Fig3.D-E
</p>
</div>
</div>
<div id="f-g.-tf-activities" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> (F-G). TF activities<a href="figure-3.html#f-g.-tf-activities" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="figure-3.html#cb22-1" tabindex="-1"></a><span class="co">#show the TF regulon of ESCC by Decouple</span></span>
<span id="cb22-2"><a href="figure-3.html#cb22-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb22-3"><a href="figure-3.html#cb22-3" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb22-4"><a href="figure-3.html#cb22-4" tabindex="-1"></a><span class="fu">library</span>(decoupleR)</span>
<span id="cb22-5"><a href="figure-3.html#cb22-5" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb22-6"><a href="figure-3.html#cb22-6" tabindex="-1"></a><span class="fu">library</span>(regioneR)</span>
<span id="cb22-7"><a href="figure-3.html#cb22-7" tabindex="-1"></a><span class="fu">library</span>(rGREAT)</span>
<span id="cb22-8"><a href="figure-3.html#cb22-8" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb22-9"><a href="figure-3.html#cb22-9" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb22-10"><a href="figure-3.html#cb22-10" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb22-11"><a href="figure-3.html#cb22-11" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb22-12"><a href="figure-3.html#cb22-12" tabindex="-1"></a><span class="fu">library</span>(ggplotify)</span>
<span id="cb22-13"><a href="figure-3.html#cb22-13" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb22-14"><a href="figure-3.html#cb22-14" tabindex="-1"></a><span class="co"># Step 1: get all regulons from human</span></span>
<span id="cb22-15"><a href="figure-3.html#cb22-15" tabindex="-1"></a>net <span class="ot">&lt;-</span> decoupleR<span class="sc">::</span><span class="fu">get_collectri</span>(<span class="at">organism=</span><span class="st">&#39;human&#39;</span>, <span class="at">split_complexes=</span><span class="cn">FALSE</span>)</span>
<span id="cb22-16"><a href="figure-3.html#cb22-16" tabindex="-1"></a>TFregul <span class="ot">&lt;-</span> net <span class="sc">%&gt;%</span> <span class="fu">pull</span>(source) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() </span>
<span id="cb22-17"><a href="figure-3.html#cb22-17" tabindex="-1"></a>regulon_gmt <span class="ot">&lt;-</span> <span class="fu">lapply</span>(TFregul,<span class="cf">function</span>(x){net <span class="sc">%&gt;%</span> <span class="fu">filter</span>(source<span class="sc">==</span>x) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(target)})</span>
<span id="cb22-18"><a href="figure-3.html#cb22-18" tabindex="-1"></a><span class="fu">names</span>(regulon_gmt)<span class="ot">&lt;-</span>TFregul</span>
<span id="cb22-19"><a href="figure-3.html#cb22-19" tabindex="-1"></a><span class="do">###write gmt</span></span>
<span id="cb22-20"><a href="figure-3.html#cb22-20" tabindex="-1"></a>write_gmt <span class="ot">&lt;-</span> <span class="cf">function</span>(x,filename) {</span>
<span id="cb22-21"><a href="figure-3.html#cb22-21" tabindex="-1"></a>    output<span class="ot">&lt;-</span><span class="fu">file</span>(filename,<span class="at">open =</span> <span class="st">&quot;wt&quot;</span>)</span>
<span id="cb22-22"><a href="figure-3.html#cb22-22" tabindex="-1"></a>    name <span class="ot">&lt;-</span> <span class="fu">names</span>(x)</span>
<span id="cb22-23"><a href="figure-3.html#cb22-23" tabindex="-1"></a>    <span class="fu">lapply</span>(name,<span class="cf">function</span>(y){</span>
<span id="cb22-24"><a href="figure-3.html#cb22-24" tabindex="-1"></a>    <span class="fu">writeLines</span>(<span class="fu">paste</span>(<span class="fu">c</span>(y,<span class="st">&quot;NA&quot;</span>,x[[y]]),<span class="at">collapse =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>),<span class="at">con =</span> output)</span>
<span id="cb22-25"><a href="figure-3.html#cb22-25" tabindex="-1"></a>     })</span>
<span id="cb22-26"><a href="figure-3.html#cb22-26" tabindex="-1"></a>    <span class="fu">close</span>(output)</span>
<span id="cb22-27"><a href="figure-3.html#cb22-27" tabindex="-1"></a>}</span>
<span id="cb22-28"><a href="figure-3.html#cb22-28" tabindex="-1"></a><span class="fu">write_gmt</span>(regulon_gmt,<span class="at">file=</span><span class="st">&quot;TFregulon.gmt&quot;</span>)</span>
<span id="cb22-29"><a href="figure-3.html#cb22-29" tabindex="-1"></a><span class="co"># Step 2: MHB</span></span>
<span id="cb22-30"><a href="figure-3.html#cb22-30" tabindex="-1"></a>ESCC_T_MHB <span class="ot">&lt;-</span> <span class="st">&quot;Data/Processed_data/MHB/Public/MHB_ESCC.bed&quot;</span></span>
<span id="cb22-31"><a href="figure-3.html#cb22-31" tabindex="-1"></a>ESCC_N_MHB <span class="ot">&lt;-</span> <span class="st">&quot;Data/Processed_data/MHB/Public/MHB_Esophagus_N.bed&quot;</span></span>
<span id="cb22-32"><a href="figure-3.html#cb22-32" tabindex="-1"></a><span class="co"># MHB gene annotation</span></span>
<span id="cb22-33"><a href="figure-3.html#cb22-33" tabindex="-1"></a>MHBGenes_TSS <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(ESCC_T_MHB,ESCC_N_MHB),<span class="cf">function</span>(y){</span>
<span id="cb22-34"><a href="figure-3.html#cb22-34" tabindex="-1"></a>              <span class="fu">toGRanges</span>(y) <span class="sc">%&gt;%</span> <span class="fu">great</span>(<span class="st">&quot;MSigDB:H&quot;</span>,<span class="st">&quot;GREAT:hg19&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">getRegionGeneAssociations</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-35"><a href="figure-3.html#cb22-35" tabindex="-1"></a>              <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span>  <span class="fu">mutate</span>(<span class="at">annotated_genes=</span><span class="fu">lapply</span>(annotated_genes,<span class="cf">function</span>(x){</span>
<span id="cb22-36"><a href="figure-3.html#cb22-36" tabindex="-1"></a>                                                x <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb22-37"><a href="figure-3.html#cb22-37" tabindex="-1"></a>                                             }) <span class="sc">%&gt;%</span> <span class="fu">do.call</span>(rbind,.) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>(),</span>
<span id="cb22-38"><a href="figure-3.html#cb22-38" tabindex="-1"></a>                                             <span class="at">dist_to_TSS =</span> <span class="fu">lapply</span>(dist_to_TSS,<span class="cf">function</span>(x){</span>
<span id="cb22-39"><a href="figure-3.html#cb22-39" tabindex="-1"></a>                                                x <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb22-40"><a href="figure-3.html#cb22-40" tabindex="-1"></a>                                             }) <span class="sc">%&gt;%</span> <span class="fu">do.call</span>(rbind,.) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb22-41"><a href="figure-3.html#cb22-41" tabindex="-1"></a>                                             <span class="fu">separate_rows</span>(annotated_genes,dist_to_TSS,<span class="at">sep=</span><span class="st">&quot;,&quot;</span>,<span class="at">convert=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-42"><a href="figure-3.html#cb22-42" tabindex="-1"></a>                                           <span class="fu">mutate</span>(<span class="at">Type=</span><span class="fu">ifelse</span>(<span class="fu">str_detect</span>(y,<span class="st">&quot;ESCC&quot;</span>),<span class="st">&quot;esophagus_T&quot;</span>,<span class="st">&quot;esophagus_N&quot;</span>))})</span>
<span id="cb22-43"><a href="figure-3.html#cb22-43" tabindex="-1"></a><span class="fu">names</span>(MHBGenes_TSS) <span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;esophagus_T&quot;</span>,<span class="st">&quot;esophagus_N&quot;</span>)</span>
<span id="cb22-44"><a href="figure-3.html#cb22-44" tabindex="-1"></a>MHBGenes_T <span class="ot">&lt;-</span> MHBGenes_TSS[[<span class="st">&quot;esophagus_T&quot;</span>]]  <span class="sc">%&gt;%</span> <span class="fu">pull</span>(annotated_genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb22-45"><a href="figure-3.html#cb22-45" tabindex="-1"></a>MHBGenes_N <span class="ot">&lt;-</span> MHBGenes_TSS[[<span class="st">&quot;esophagus_N&quot;</span>]]  <span class="sc">%&gt;%</span> <span class="fu">pull</span>(annotated_genes) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb22-46"><a href="figure-3.html#cb22-46" tabindex="-1"></a></span>
<span id="cb22-47"><a href="figure-3.html#cb22-47" tabindex="-1"></a><span class="co"># step 2.1 rGREAT of regulon regulated by MHB</span></span>
<span id="cb22-48"><a href="figure-3.html#cb22-48" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;regulon_MHB&quot;</span>))</span>
<span id="cb22-49"><a href="figure-3.html#cb22-49" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;regulon_MHB&quot;</span>)</span>
<span id="cb22-50"><a href="figure-3.html#cb22-50" tabindex="-1"></a><span class="co"># Tumor</span></span>
<span id="cb22-51"><a href="figure-3.html#cb22-51" tabindex="-1"></a><span class="cf">for</span> ( i <span class="cf">in</span> <span class="fu">names</span>(regulon_gmt)){</span>
<span id="cb22-52"><a href="figure-3.html#cb22-52" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> regulon_gmt[[i]] </span>
<span id="cb22-53"><a href="figure-3.html#cb22-53" tabindex="-1"></a>    regulon_MHB<span class="ot">&lt;-</span> MHBGenes_TSS[[<span class="st">&quot;esophagus_T&quot;</span>]] <span class="sc">%&gt;%</span> <span class="fu">filter</span>(annotated_genes <span class="sc">%in%</span> gene)</span>
<span id="cb22-54"><a href="figure-3.html#cb22-54" tabindex="-1"></a>    <span class="fu">write.table</span>(regulon_MHB, <span class="at">file=</span><span class="fu">paste0</span>(<span class="st">&quot;regulon_MHB/&quot;</span>,i,<span class="st">&quot;_regulon_targets_MHB_T.bed&quot;</span>),<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb22-55"><a href="figure-3.html#cb22-55" tabindex="-1"></a>}</span>
<span id="cb22-56"><a href="figure-3.html#cb22-56" tabindex="-1"></a><span class="co"># step 2.2 do rGREAT of regulon in MHBGene_T</span></span>
<span id="cb22-57"><a href="figure-3.html#cb22-57" tabindex="-1"></a>regulons <span class="ot">&lt;-</span> <span class="fu">read_gmt</span>(<span class="st">&quot;TFregulon.gmt&quot;</span>, </span>
<span id="cb22-58"><a href="figure-3.html#cb22-58" tabindex="-1"></a>    <span class="at">from =</span> <span class="st">&quot;SYMBOL&quot;</span>, <span class="at">to =</span> <span class="st">&quot;ENTREZ&quot;</span>, <span class="at">orgdb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>) </span>
<span id="cb22-59"><a href="figure-3.html#cb22-59" tabindex="-1"></a>regulons_MHB_enrichment <span class="ot">&lt;-</span> <span class="fu">toGRanges</span>(ESCC_T_MHB) <span class="sc">%&gt;%</span> <span class="fu">great</span>(regulons, <span class="st">&quot;hg19&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-60"><a href="figure-3.html#cb22-60" tabindex="-1"></a>                            <span class="fu">getEnrichmentTable</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_value_hyper<span class="sc">&lt;=</span><span class="fl">0.05</span>)</span>
<span id="cb22-61"><a href="figure-3.html#cb22-61" tabindex="-1"></a><span class="co"># saving </span></span>
<span id="cb22-62"><a href="figure-3.html#cb22-62" tabindex="-1"></a><span class="fu">write.table</span>(regulons_MHB_enrichment,<span class="at">file=</span><span class="st">&quot;regulons_MHB_enrichment_significant.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb22-63"><a href="figure-3.html#cb22-63" tabindex="-1"></a><span class="co"># Step 3: RNA-seq  limma</span></span>
<span id="cb22-64"><a href="figure-3.html#cb22-64" tabindex="-1"></a><span class="co"># ESCC Exp &amp;&amp; DEGs</span></span>
<span id="cb22-65"><a href="figure-3.html#cb22-65" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Data/Processed_data/ESCC/ESCC_log2TPM_Wilcox.RData&quot;</span>)</span>
<span id="cb22-66"><a href="figure-3.html#cb22-66" tabindex="-1"></a>pheno_data <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="st">&quot;samples&quot;</span><span class="ot">=</span><span class="fu">names</span>(ESCC_log2TPM),</span>
<span id="cb22-67"><a href="figure-3.html#cb22-67" tabindex="-1"></a>                         <span class="st">&quot;Group&quot;</span><span class="ot">=</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(ESCC_log2TPM),<span class="st">&quot;_&quot;</span>),<span class="cf">function</span>(x) x[<span class="dv">2</span>]))</span>
<span id="cb22-68"><a href="figure-3.html#cb22-68" tabindex="-1"></a>group <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">substr</span>(pheno_data<span class="sc">$</span>Group,<span class="dv">1</span>,<span class="dv">1</span>),<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;N&quot;</span>,<span class="st">&quot;T&quot;</span>))  <span class="do">### tumor or normal</span></span>
<span id="cb22-69"><a href="figure-3.html#cb22-69" tabindex="-1"></a>orign <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">substr</span>(pheno_data<span class="sc">$</span>Group,<span class="dv">2</span>,<span class="fu">nchar</span>(pheno_data<span class="sc">$</span>Group)))  <span class="do">##Paired </span></span>
<span id="cb22-70"><a href="figure-3.html#cb22-70" tabindex="-1"></a><span class="do">##############</span></span>
<span id="cb22-71"><a href="figure-3.html#cb22-71" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group<span class="sc">+</span>orign)</span>
<span id="cb22-72"><a href="figure-3.html#cb22-72" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">lmFit</span>(ESCC_log2TPM,design)</span>
<span id="cb22-73"><a href="figure-3.html#cb22-73" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">eBayes</span>(fit)</span>
<span id="cb22-74"><a href="figure-3.html#cb22-74" tabindex="-1"></a><span class="fu">options</span>(<span class="at">digits =</span> <span class="dv">4</span>)</span>
<span id="cb22-75"><a href="figure-3.html#cb22-75" tabindex="-1"></a>deg <span class="ot">&lt;-</span>  <span class="fu">topTable</span>(fit,<span class="at">coef=</span><span class="dv">2</span>,<span class="at">adjust=</span><span class="st">&#39;BH&#39;</span>,<span class="at">number =</span> <span class="dv">30000</span>,<span class="at">sort.by =</span> <span class="st">&#39;logFC&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-76"><a href="figure-3.html#cb22-76" tabindex="-1"></a>      <span class="fu">select</span>(logFC, t, P.Value) <span class="sc">%&gt;%</span> </span>
<span id="cb22-77"><a href="figure-3.html#cb22-77" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(t)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-78"><a href="figure-3.html#cb22-78" tabindex="-1"></a>        <span class="fu">as.matrix</span>()</span>
<span id="cb22-79"><a href="figure-3.html#cb22-79" tabindex="-1"></a><span class="co"># step 4.  TF Activity inference with Univariate Linear Model (ULM)</span></span>
<span id="cb22-80"><a href="figure-3.html#cb22-80" tabindex="-1"></a>sample_acts <span class="ot">&lt;-</span> <span class="fu">run_ulm</span>(<span class="at">mat=</span>ESCC_log2TPM, <span class="at">net=</span>net, <span class="at">.source=</span><span class="st">&#39;source&#39;</span>, <span class="at">.target=</span><span class="st">&#39;target&#39;</span>,</span>
<span id="cb22-81"><a href="figure-3.html#cb22-81" tabindex="-1"></a>                  <span class="at">.mor=</span><span class="st">&#39;mor&#39;</span>, <span class="at">minsize =</span> <span class="dv">5</span>) </span>
<span id="cb22-82"><a href="figure-3.html#cb22-82" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb22-83"><a href="figure-3.html#cb22-83" tabindex="-1"></a>n_tfs <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb22-84"><a href="figure-3.html#cb22-84" tabindex="-1"></a>sample_acts_mat <span class="ot">&lt;-</span> sample_acts <span class="sc">%&gt;%</span></span>
<span id="cb22-85"><a href="figure-3.html#cb22-85" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">&#39;condition&#39;</span>, <span class="at">names_from =</span> <span class="st">&#39;source&#39;</span>,</span>
<span id="cb22-86"><a href="figure-3.html#cb22-86" tabindex="-1"></a>              <span class="at">values_from =</span> <span class="st">&#39;score&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-87"><a href="figure-3.html#cb22-87" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&#39;condition&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-88"><a href="figure-3.html#cb22-88" tabindex="-1"></a>  <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb22-89"><a href="figure-3.html#cb22-89" tabindex="-1"></a></span>
<span id="cb22-90"><a href="figure-3.html#cb22-90" tabindex="-1"></a><span class="co"># Get top 50 tfs with more variable means across clusters</span></span>
<span id="cb22-91"><a href="figure-3.html#cb22-91" tabindex="-1"></a>regulon_mhb_tfs <span class="ot">&lt;-</span> regulons_MHB_enrichment  <span class="sc">%&gt;%</span> <span class="fu">pull</span>(id) <span class="sc">%&gt;%</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(sample_acts_mat))</span>
<span id="cb22-92"><a href="figure-3.html#cb22-92" tabindex="-1"></a>tfs <span class="ot">&lt;-</span> sample_acts <span class="sc">%&gt;%</span></span>
<span id="cb22-93"><a href="figure-3.html#cb22-93" tabindex="-1"></a>          <span class="fu">filter</span>(source <span class="sc">%in%</span> regulon_mhb_tfs ) <span class="sc">%&gt;%</span>     <span class="do">#### only contains the MHB enriched regulons</span></span>
<span id="cb22-94"><a href="figure-3.html#cb22-94" tabindex="-1"></a>          <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span></span>
<span id="cb22-95"><a href="figure-3.html#cb22-95" tabindex="-1"></a>          <span class="fu">summarise</span>(<span class="at">std =</span> <span class="fu">sd</span>(score)) <span class="sc">%&gt;%</span></span>
<span id="cb22-96"><a href="figure-3.html#cb22-96" tabindex="-1"></a>          <span class="fu">arrange</span>(<span class="sc">-</span><span class="fu">abs</span>(std)) <span class="sc">%&gt;%</span>          </span>
<span id="cb22-97"><a href="figure-3.html#cb22-97" tabindex="-1"></a>          <span class="fu">head</span>(n_tfs) <span class="sc">%&gt;%</span></span>
<span id="cb22-98"><a href="figure-3.html#cb22-98" tabindex="-1"></a>          <span class="fu">pull</span>(source)</span>
<span id="cb22-99"><a href="figure-3.html#cb22-99" tabindex="-1"></a>sample_acts_mat <span class="ot">&lt;-</span> sample_acts_mat[,tfs]</span>
<span id="cb22-100"><a href="figure-3.html#cb22-100" tabindex="-1"></a><span class="co"># Scale per sample</span></span>
<span id="cb22-101"><a href="figure-3.html#cb22-101" tabindex="-1"></a>sample_acts_mat_scale <span class="ot">&lt;-</span> <span class="fu">scale</span>(sample_acts_mat)</span>
<span id="cb22-102"><a href="figure-3.html#cb22-102" tabindex="-1"></a><span class="co"># Ranking</span></span>
<span id="cb22-103"><a href="figure-3.html#cb22-103" tabindex="-1"></a>Tx <span class="ot">&lt;-</span> sample_acts_mat <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-104"><a href="figure-3.html#cb22-104" tabindex="-1"></a>                           <span class="fu">mutate</span>(<span class="at">T=</span><span class="fu">rowMeans</span>(<span class="fu">across</span>(<span class="fu">contains</span>(<span class="st">&quot;_T&quot;</span>)),<span class="at">na.rm=</span>T),</span>
<span id="cb22-105"><a href="figure-3.html#cb22-105" tabindex="-1"></a>                                  <span class="at">N=</span><span class="fu">rowMeans</span>(<span class="fu">across</span>(<span class="fu">contains</span>(<span class="st">&quot;_N&quot;</span>)),<span class="at">na.rm=</span>T),</span>
<span id="cb22-106"><a href="figure-3.html#cb22-106" tabindex="-1"></a>                                  <span class="at">cluster=</span><span class="fu">ifelse</span>(T<span class="sc">&gt;</span>N,<span class="st">&quot;1&quot;</span>,<span class="st">&quot;2&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-107"><a href="figure-3.html#cb22-107" tabindex="-1"></a>                          <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;name&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(cluster,name)</span>
<span id="cb22-108"><a href="figure-3.html#cb22-108" tabindex="-1"></a>gene_order <span class="ot">&lt;-</span> Tx <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb22-109"><a href="figure-3.html#cb22-109" tabindex="-1"></a>sample_acts_mat_scale <span class="ot">&lt;-</span> sample_acts_mat_scale[,gene_order]</span>
<span id="cb22-110"><a href="figure-3.html#cb22-110" tabindex="-1"></a><span class="co"># Choose color palette</span></span>
<span id="cb22-111"><a href="figure-3.html#cb22-111" tabindex="-1"></a>palette_length <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb22-112"><a href="figure-3.html#cb22-112" tabindex="-1"></a>my_color <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;white&quot;</span>,<span class="st">&quot;firebrick&quot;</span>))(palette_length)</span>
<span id="cb22-113"><a href="figure-3.html#cb22-113" tabindex="-1"></a></span>
<span id="cb22-114"><a href="figure-3.html#cb22-114" tabindex="-1"></a>my_breaks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">0</span>, <span class="at">length.out=</span><span class="fu">ceiling</span>(palette_length<span class="sc">/</span><span class="dv">2</span>) <span class="sc">+</span> <span class="dv">1</span>),</span>
<span id="cb22-115"><a href="figure-3.html#cb22-115" tabindex="-1"></a>               <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">3</span>, <span class="at">length.out=</span><span class="fu">floor</span>(palette_length<span class="sc">/</span><span class="dv">2</span>)))</span>
<span id="cb22-116"><a href="figure-3.html#cb22-116" tabindex="-1"></a></span>
<span id="cb22-117"><a href="figure-3.html#cb22-117" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb22-118"><a href="figure-3.html#cb22-118" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(<span class="st">&quot;Data/Figures/Fig3/Fig3.F.pdf&quot;</span>),<span class="at">width=</span><span class="dv">4</span>,<span class="at">height=</span><span class="dv">8</span>)</span>
<span id="cb22-119"><a href="figure-3.html#cb22-119" tabindex="-1"></a>p<span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(<span class="fu">t</span>(sample_acts_mat_scale), </span>
<span id="cb22-120"><a href="figure-3.html#cb22-120" tabindex="-1"></a>              <span class="at">cluster_col=</span>F, </span>
<span id="cb22-121"><a href="figure-3.html#cb22-121" tabindex="-1"></a>              <span class="at">cluster_row=</span>F,</span>
<span id="cb22-122"><a href="figure-3.html#cb22-122" tabindex="-1"></a>              <span class="at">border_color =</span> <span class="cn">NA</span>, <span class="at">color=</span>my_color, <span class="at">breaks =</span> my_breaks)</span>
<span id="cb22-123"><a href="figure-3.html#cb22-123" tabindex="-1"></a><span class="fu">print</span>(p)</span>
<span id="cb22-124"><a href="figure-3.html#cb22-124" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-40"></span>
<img src="Data/Figures/Fig3/Fig3.F.png" alt="Fig3.F" width="80%" id="fig-unnamed-chunk-40-1" />
<p class="caption">
 3.3: Fig3.F
</p>
</div>
<div class="sourceCode" id="cb23"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb23-1"><a href="figure-3.html#cb23-1" tabindex="-1"></a><span class="co"># KLF5, SOX2 TF ChIP-seq profiles in ESCC cell line.</span></span>
<span id="cb23-2"><a href="figure-3.html#cb23-2" tabindex="-1"></a><span class="va">bw_path</span><span class="op">=</span><span class="st">&quot;ChIP/ESCC_TF/&quot;</span></span>
<span id="cb23-3"><a href="figure-3.html#cb23-3" tabindex="-1"></a><span class="va">bw1</span><span class="op">=</span><span class="va">$1</span></span>
<span id="cb23-4"><a href="figure-3.html#cb23-4" tabindex="-1"></a><span class="va">tag</span><span class="op">=</span><span class="va">$2</span></span>
<span id="cb23-5"><a href="figure-3.html#cb23-5" tabindex="-1"></a><span class="va">MHB_TS</span><span class="op">=</span><span class="st">&quot;ChIP/profiles/DB/MHB_TS.bed&quot;</span></span>
<span id="cb23-6"><a href="figure-3.html#cb23-6" tabindex="-1"></a><span class="va">MHB_NS</span><span class="op">=</span><span class="st">&quot;ChIP/profiles/DB/MHB_NS.bed&quot;</span></span>
<span id="cb23-7"><a href="figure-3.html#cb23-7" tabindex="-1"></a><span class="va">MHB_Common</span><span class="op">=</span><span class="st">&quot;ChIP/profiles/DB/MHB_NT_Common.bed&quot;</span></span>
<span id="cb23-8"><a href="figure-3.html#cb23-8" tabindex="-1"></a><span class="va">BG</span><span class="op">=</span><span class="st">&quot;ChIP/profiles/DB/ESCC_MHB_random.bed&quot;</span></span>
<span id="cb23-9"><a href="figure-3.html#cb23-9" tabindex="-1"></a><span class="co"># matrix</span></span>
<span id="cb23-10"><a href="figure-3.html#cb23-10" tabindex="-1"></a><span class="ex">computeMatrix</span> reference-point  <span class="at">--referencePoint</span> center  <span class="at">-S</span> <span class="va">${bw_path}</span>/<span class="va">$bw1</span> <span class="dt">\</span></span>
<span id="cb23-11"><a href="figure-3.html#cb23-11" tabindex="-1"></a>                            <span class="at">-R</span> <span class="va">${MHB_TS}</span> <span class="va">${MHB_NS}</span> <span class="va">${MHB_Common}</span> <span class="va">${BG}</span> <span class="dt">\</span></span>
<span id="cb23-12"><a href="figure-3.html#cb23-12" tabindex="-1"></a>                            <span class="at">-b</span> 2000 <span class="at">-a</span> 2000 <span class="at">-bs</span> 40 <span class="at">--skipZeros</span> <span class="dt">\</span></span>
<span id="cb23-13"><a href="figure-3.html#cb23-13" tabindex="-1"></a>                            <span class="at">--smartLabels</span> <span class="at">-p</span> 20  <span class="dt">\</span></span>
<span id="cb23-14"><a href="figure-3.html#cb23-14" tabindex="-1"></a>                            <span class="at">--outFileNameMatrix</span> profiles/ESCC_<span class="va">${tag}</span>_MHB.tab <span class="dt">\</span></span>
<span id="cb23-15"><a href="figure-3.html#cb23-15" tabindex="-1"></a>                            <span class="at">-o</span> profiles/ESCC_<span class="va">${tag}</span>_MHB.gz </span>
<span id="cb23-16"><a href="figure-3.html#cb23-16" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb23-17"><a href="figure-3.html#cb23-17" tabindex="-1"></a><span class="ex">plotHeatmap</span> <span class="at">-m</span> profiles/ESCC_<span class="va">${tag}</span>_MHB.gz <span class="dt">\</span></span>
<span id="cb23-18"><a href="figure-3.html#cb23-18" tabindex="-1"></a>    <span class="at">--missingDataColor</span> white <span class="dt">\</span></span>
<span id="cb23-19"><a href="figure-3.html#cb23-19" tabindex="-1"></a>     <span class="at">-out</span> profiles/ESCC_<span class="va">${tag}</span>_MHB_heatprofiles.pdf <span class="dt">\</span></span>
<span id="cb23-20"><a href="figure-3.html#cb23-20" tabindex="-1"></a>     <span class="at">--colorMap</span> Purples <span class="dt">\</span></span>
<span id="cb23-21"><a href="figure-3.html#cb23-21" tabindex="-1"></a>     <span class="at">--heatmapHeight</span> 12 <span class="at">--heatmapWidth</span> 3</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-42"></span>
<img src="Data/Figures/Fig3/Fig3.G.KLF5.png" alt="Fig3.G" width="40%" id="fig-unnamed-chunk-42-1" /><img src="Data/Figures/Fig3/Fig3.G.SOX2.png" alt="Fig3.G" width="40%" id="fig-unnamed-chunk-42-2" />
<p class="caption">
 3.4: Fig3.G
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-literature.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
